\linespread{1.0}

\chapter{General Introduction}
%--------------------------------------------------------- Sec

\subsection{The Problem}

Biodiversity loss currently threatens the diversity of life on Earth. It is estimated by the United Nations Convention on Biological Diversity (CBD) in their Global Biodiversity Outlook report that, of the estimated eight million species known, over one million animal and plant species currently face risk of extinction in the next few decades due solely to increased anthropogenic activities \cite{cbd2020global}. This troubling revelation is made all the more real since the majority of species still await discovery and formal description. 

\vspace{5mm}

Through traditional means of morphological identification, taxonomists have \\ painstakingly managed to categorize just over 1 million species in the last 250 years alone. DNA barcoding \cite{hebert2003biological, hebert2003barcoding}, proposed nearly 20 years ago in 2003 as a viable solution to the taxonomic impediment, has since revolutionized the way Linnean taxonomy is done. The premise of DNA barcoding is quite straightforward. The technique proposes to make accurate and rapid species diagnoses through leveraging easily obtained genetic variation seen in short molecular DNA gene regions collected from unknown specimens of interest. In animals, DNA barcoding specifically employs the cytochrome \textit{c} oxidase subunit I (COI) gene found in the mitochondria of cells, which is highly abundant and found in all animal species. Further, while DNA barcoding's primary goal is to facilitate the acceleration of specimen identification and species discovery, a number of uses and applications outside of biodiversity science have been brought forth. In particular, government regulatory bodies worldwide such as the Canadian Food Inspection Agency and Agriculture and Agri-Food Canada to name a few have harnessed the true power DNA barcoding has to offer in the combatting of systemic seafood fraud, as well as in the monitoring of the impacts and spread of invasive species on natural ecosystems.

\vspace{5mm}

Robust estimation of adequate specimen sample sizes for DNA-based species \\ identification of animal taxa through DNA barcoding is central to timely biodiversity \\ conservation and management. However, this problem is fraught with myriad challenges including species rarity and project costs \cite{cameron2006will, stein2014is}. Further, because species show \\ remarkable genomic marker variation and rates of molecular evolution within and among taxa, along with differing evolutionary and life histories, knowing how many specimens of a given species likely need to be collected to observe the majority of existing genetic diversity present within animal species of interest to biodiversity researchers and regulatory scientists is a difficult question to answer. While practical sample sizes for DNA barcoding typically range from 5-10 specimens per species \cite{zhang2010estimating}, anywhere from a single individual to hundreds of specimens may be targeted depending on the study \cite{hajibabaei2007dna, matz2005likelihood, zhang2010estimating}. \\ Unfortunately, little work has been done to determine optimal sampling depths in a \\ statistically rigorous manner. 

The majority of studies conducted to date on estimating sample sizes for DNA \\ barcoding have employed sophisticated parametric statistical models having strong \\ underlying assumptions. Unfortunately, the success of this approach is highly dependent on the taxa and molecular genetic loci being considered. This warrants the introduction of more general, user-friendly approaches applicable to wide-ranging taxonomic groups and molecular marker genes. 

\vspace{5mm}

\subsection{Thesis Overview} \label{sec:intro1}

This thesis outlines a novel statistical framework for assessment of COI DNA barcode haplotype sampling completeness. Much of the work presented herein reflects a spin-off of research completed at the Master's level in bioinformatics.  

\vspace{5mm}

In Chapter 2, existing literature on sample size determination for DNA barcoding is first reviewed. Here, evidence points to a large knowledge gap in statistical and computational methods currently available for this task. Specifically, too much focus has been placed on inflexible parametric models rather than generalized flexible ones. Further this work finds that efforts have been improperly delegated to sampling as many species as possible, rather than maximizing the number of specimens collected. A case study on ray-finned fishes retrieved from the Barcode of Life Data Systems \cite{ratnasingham2007bold} clearly highlights this shortcoming, along with the need for the development of approaches which incorporate more 
\\ species-level information. 

\vspace{5mm}

Chapter 3 builds on fundamental concepts of evolutionary biology and statistics \\ introduced and outlined in Chapter 2 through detailing a novel nonparametric stochastic local search optimization algorithm in the R statistical programming language to better address the need for improved sampling strategies for DNA barcoding initiatives. The method, called {\tt HACSim} (short for \textbf{H}aplotype \textbf{A}ccumulation \textbf{C}urve \textbf{Sim}ulator) available as an R package for global use, employs easily obtainable genomic information from a sample of previously-assembled species-specific DNA sequence alignments and is tested on a variety of hypothetical and real species mined from the Barcode of Life Data Systems (BOLD). Specifically, the method employs interation and randomness to extrapolate \\ species' haplotype accumulation curves toward an asymptote to assess where such curves may level off. The approach is found to work well for a number of relevant species, consistently suggesting that hundreds to thousands of specimens are actually needed to be randomly sampled across their geographic and ecologic ranges to be confident that much species-level genomic variation has been sufficiently captured.  


\vspace{5mm}

Chapter 4 extends elements discussed in Chapter 3 through delving further into the useability and applicability of {\tt HACSim} via a detailed statistical simulation study to assess both the validity and overall performance of {\tt HACSim} and its utility for assessing \\ intraspecific sampling completeness within DNA barcoding studies for a variety of species mined from the Barcode of Life Data Systems (BOLD) \cite{ratnasingham2007bold}. {\tt HACSim} is demonstrated to possess good statistical properties, including high consistency between successive 
\\ algorithm runs and high coverage probabilities for desired capture of intraspecific \\ haplotype variation.

\vspace{5mm}

Finally, in Chapter 5, it is argued that DNA barcoding is currently lacking in statistical rigor and that better statistical methods are necessary to more accurately assess standing genetic variation at the species level when it comes to estimating the DNA barcode gap. The use {\tt HACSim} is suggested to address the problem of improper allocation of specimen sampling effort. Kernel density estimation plots, along with quadrant plots, are advocated for in place of traditional histograms to more easily detect outlier and problematic taxa that reflect potential failures of DNA barcoding. Hypothesis testing, in addition to \\ nonparametric bootstrapping are recommended to place DNA barcoding and barcode gap analyses on firmer statistical ground through estimation of confidence intervals of \\ intraspecific and interspecific genetic distances. All proposed approaches are illustrated through a case study focussing on Pacific fishes of Canada \cite{steinke2009dna}.

\vspace{5mm}

\subsection{Thesis Statement}

Through the development of a novel stochastic simulation algorithm for the generation of haplotype accumulation curves, the current research will provide a framework that can be employed to determine plausible specimen sample sizes sufficient to quantify levels of haplotypic sampling completeness within species under both uniform and non-uniform haplotype frequency distributions. Such a framework will be valuable in promoting a greater degree of statistical thoroughness in future DNA barcoding studies. 

\vspace{5mm}

\subsection{Statement of Contributions}
  
All chapters presented in this thesis are original and were the sole effort of JDP, \\ including review of the primary literature, conceptualization of ideas, implementation of code, design of experiments and writing of individual manuscripts. All other coauthors either assisted directly with writing of code and/or running of experiments or participated in the editing of final manuscript versions.

\vspace{5mm}

The following articles are published, under review or in preparation

\begin{itemize}

\item Phillips, J.D., Gillis, D.J. and Hanner, R.H. (2019). Incomplete estimates of genetic diversity within species: Implications for DNA barcoding. \textit{Ecology and Evolution}, \textbf{9}(5): 2996-3010. 

\vspace{1mm}

\item Phillips, J.D., French, S.H., Hanner, R.H. and Gillis, D.J. (2020). HACSim: An R package to estimate intraspecific sample sizes for genetic diversity assessment using haplotype accumulation curves. \textit{PeerJ Computer Science}, \textbf{6}(192): 1-37.

\vspace{1mm}

\item Phillips, J.D., Bootsma, S.E., Hanner, R.H. and Gillis, D.J. (\textit{In preparation}). \\ Solving the genetic specimen sample size problem for DNA barcoding with a local search optimization algorithm.

\vspace{1mm}

\item Phillips, J.D., Gillis, D.J. and Hanner, R.H. (\textit{In preparation}). Lack of statistical rigor in DNA barcoding likely invalidates the presence of a true species' barcode gap.

\end{itemize} 