\linespread{1.0}

\chapter{Lack of statistical rigor in DNA barcoding likely invalidates the presence of a true species' barcode gap}

\noindent{Jarrett D. Phillips$^{1 *}$, Daniel J. Gillis$^1$, Robert H. Hanner$^{2, 3}$ \\ $^1$School of Computer Science \\ $^2$Biodiversity Institute of Ontario \\ $^3$Department of Integrative Biology}

\vspace{\fill}

\noindent{This chapter is in preparation for submission to \textit{Frontiers in Ecology and Evolution}}

\section{Prologue}

Here, evidence is presented to support a lack of statistical rigor in DNA barcoding as it pertains to sound estimation of the DNA barcode gap -- the difference between genetic variation observed within and between species. 



Biodiversity researchers and regulatory scientists alike are called on to incorporate more advanced, yet accessible, statistical methods to better characterize standing levels of haplotype variation at the species level. These include employing {\tt HACSim} to estimate likely required specimen sample sizes for barcoding studies, generating kernel density estimation plots instead of histograms, along with other underused graphics to depict the barcode gap (or lack thereof), and utilizing nonparametric bootstrapping to produce point and interval estimates of quantities central to DNA barcoding and barcode gap analyses.



This manuscript is in preparation for submission to \textit{Frontiers in Ecology and Evolution}.  

\begin{abstract}

DNA barcoding was first conceived at the start of the 21st century as a means to rapidly characterize Earth's dwindling biodiversity faster than traditional taxonomy ever could over the last 250 years, solely on the basis of sequence variation found in a single gene or a few universal gene markers. The proposal was met with both high praise and stark criticism. Within the first few years of barcoding's introduction, what followed was the polarization of the biodiversity community-at-large, the sparking of heated discussion among researchers, and the incitement of widespread debate amidst established schools of biological thought. Much initial backlash stemmed from the conflation of DNA barcoding's presumed role as a potent tool for specimen identification, as well as one for species discovery. Flashing forward almost two decades later, the application of DNA barcoding for identification purposes is a now largely settled argument, whereas its promise for  delineating species still remains controversial. Nevertheless, DNA barcoding has proved itself to be both fully capable of rising to the challenge and highly resilient to change. However, the story does not end here. As reference sequence libraries continue to grow exponentially in size, there is now the need to identify novel ways of meaningfully analyzing vast amounts of available DNA barcode data.



Here, it is demonstrated that the interpretation of DNA barcoding data is lacking in statistical rigor. To highlight this, focus is set specifically on one key concept that has become a household name in the field: the DNA barcode gap. Arguments outlined herein specifically centre on DNA barcoding in animal taxa and stem from three angles: (1) the improper allocation of specimen sampling effort necessary to capture adequate levels of within-species genetic variation, (2) failing to properly visualize intraspecific and \\ interspecific genetic distances, and (3) the inconsistent, inappropriate use, or absence of statistical inferential procedures in DNA barcoding gap analyses. Furthermore, simple statistical solutions are outlined which can greatly propel the use of DNA barcoding as a tool to irrefutably match unknowns to knowns on the basis of the barcoding gap with a high degree of confidence. Proposed methods examined herein are illustrated through application to DNA barcode sequence data from Canadian Pacific fish species as a case study.

\end{abstract}


\section{Introduction}

\subsection{DNA Barcoding: Historical Development}

In its infancy, DNA barcoding \cite{hebert2003biological}  was envisaged as a means to resolve a longstanding problem facing biodiversity science: the taxonomic impediment. 
Significant lack of \\ progress in taxon characterization due to an aging expert population, in conjunction with the limited availability of adequate research funding to support such endeavors, has greatly \\ contributed to the lack of timely dissemination of taxonomic information and knowledge in recent decades \cite{cao2016rapid}. In a time marked by global species extinction and ongoing \\ environmental crisis, DNA barcoding promised to accelerate the description of novel taxa, as well as revise the status of existing ones. This was possible through the assembly of genetic ``signatures" within a centralized repository more rapidly than customary Linnean classification was even capable. The proposal seemed, at first, like wishful thinking within some academic circles. DNA barcoding employs short molecular sequence tags from \\ standardized genomic regions, such as the \textit{c.}~650 bp fragment from the 5' end of the mitochondrial cytochrome \textit{c} oxidase subunit I (COI) gene in animals, to establish \\ taxon-level matches to unknown specimen queries at any life stage across the animal \\ kingdom \cite{hebert2003barcoding}. Several past studies have also discussed the appeal of isolating a barcode sequence from mitochondrial DNA (mtDNA), as opposed to its nuclear counterpart (\textit{e.g.}, \cite{hubert2015dna, hubert2008identifying}). Notably, due to the effects of random genetic drift, mutational substitutions are expected to reach fixation within populations and become diagnostic of species sooner because the effective population size of mitochondria is one-quarter less than that of nuclear genes. The specific choice of COI as the currently unattested DNA barcode for animals is justified in several respects: (1) that it is protein-coding, (2) that it possesses a reasonably high rate of  nucleotide substitution, and (3) that it lacks introns and comprises few \\ insertions/deletions (indels) and no stop codons. In addition to these desirable \\ characteristics, its ease of amplification, sequencing and alignment across most taxa, due to its highly conserved nature, makes COI the preferred gene marker over other such loci that meet only some of the abovementioned requirements \cite{phillips2019incomplete}. Despite this, the attractiveness of DNA barcoding's use as both a specimen identification and species discovery tool is fraught with much controversy. The molecular identification of specimens to any level of biological organization through DNA barcodes necessarily depends \textit{a priori} on known taxon designations brought about through the current state of taxonomic practices \cite{desalle2006species}. \\ Furthermore, DNA barcoding's success rests crucially on extant species-level haplotype diversity as well as the distinction between intraspecific and interspecific genetic variation across taxa \cite{phillips2019incomplete}. When such links can be established, observable geographic and historic patterns of genetic diversity are then readily explained.



While DNA barcoding has found myriad applications in diverse subdisciplines of \\ evolutionary biology, ecology, and more broadly in biodiversity science, one surprising area, namely applied regulatory forensics, has reaped the benefits barcoding has to offer in \\ unparalleled ways throughout the years. The identification of regulated species of \\ socioeconomic importance through the accumulation of DNA barcodes has been \\ instrumental in combatting instances of seafood market fraud as well as monitoring the introduction and spread of invasive pests, particularly in Canada and the USA \\ (\textit{e.g.}, seafood: Shehata \textit{et al.}~\cite{shehata2018dna}, Shehata \textit{et al.}~\cite{shehata2019survey}; meat products: Naaum \textit{et al.}~\cite{naaum2018complementary}, Shehata \textit{et al.}~\cite{shehata2019revisiting}; invasive arthropods: Madden \textit{et al.}~\cite{madden2019using}). Despite this, several obstacles still remain. The inherent dynamism characteristic of public genomic databases, such as the Barcode of Life Data Systems (BOLD; Ratnasingham and Hebert \cite{ratnasingham2007bold}) \\ (http://www.boldsystems.org) and GenBank, precludes their routine use for such a task. The fact that the addition of new specimen records to community databanks may produce contradictory findings over time is problematic \cite{phillips2020hacsim}. Instead, regulatory sequence \\ databases should be populated with static taxon records traced to voucher specimens \\ whenever possible so that such issues can be mitigated. While the inclusion of \\ fit-for-purpose DNA sequences in governmental repositories like the European and \\ Mediterranean Plant Protection Organization (EPPO)'s Q-bank (https://qbank.eppo.int) for agricultural/quarantine pests and the United States Food and Drug Administration \\ (USFDA)'s Reference Standard Sequence Library for Seafood Identification (RSSL) \\ (https://www.fda.gov/food/dna-based-seafood-identification/reference-standard-sequence-\\library-seafood-identification-rssl) for seafood species represents a step in the right \\ direction, small sample size issues continue to plague the arena \cite{phillips2020hacsim}. Further, the deep sampling of an adequate number of specimens necessary to capture sufficient levels of standing haplotype variation within species is critical if high confidence in specimen \\ assignments is desired \cite{doorenweerd2020highly, phillips2019incomplete, phillips2020hacsim, phillips2015exploration}. This is even more important since much sequence data in BOLD and GenBank suffer from serious quality control issues (namely biological and/or methodological). To ensure accurate identification, access to fewer \\ high-quality specimen records is always preferable to the availability of many problematic records \cite{collins2014known}. DNA-based identification accomplished through DNA barcoding places \\ heavy reliance on the accuracy and completeness of reference sequence libraries to enable the rapid assignment of unknown specimens to valid or putative species, depending on whether the ultimate goal is specimen identification or species discovery respectively.  As use of distance-based methods strongly outweighs other taxon identification approaches (\textit{e.g.}, tree-based algorithms \cite{barbera2019epang} or character-based approaches (\textit{e.g.}, \cite{rach2008character, wong2009identifying}) within most DNA barcoding studies, a means of directly testing the overall performance of DNA barcoding is needed. Such a path forward is provided by the DNA barcode gap.  



\subsection{DNA Barcoding and the Barcode Gap: A Perfect Harmony?}

A well-established tenet in the field is that the majority of DNA barcode variation found across species exceeds genetic variability seen within species. This apparent ``barcode gap" \cite{meyer2005dna} was recognized early on as a critical factor to the success of DNA barcoding. The existence of a species' barcoding gap is often invoked as evidence that DNA barcoding ``works" in practice \cite{stoeckle2014dna}. Under current sampling efforts and morphological \\ identifications associated with DNA barcodes, a large number of species show greater than 2\% genetic distance to their nearest heterospecific and typically exhibit less than 1\% intraspecific distance \cite{hubert2015dna}. 



Researchers have differing opinions on how the DNA barcode gap ought to be defined. Whereas Meyer and Paulay \cite{meyer2005dna} advocated the use of the mean genetic distance to the nearest neighbour, employment of the minimum interspecific distance is now \\ commonplace. Reliance on the former metric tends to exaggerate the presence of a real species barcoding gap through inflating false positives, leading to misidentification of \\ specimens \cite{meier2008use}. Thus, on the basis of this paradigm shift, it is wholly conceivable that published DNA barcoding studies have likely reported biased estimates of the barcoding gap at the species level and therefore warrant revisitation and cautious interpretation. \\ Perhaps this is why the barcoding gap is depicted using both the mean and maximum intraspecific genetic distance within BOLD's Barcode Gap Analysis tool available in the user Workbench. Meyer and Paulay \cite{meyer2005dna} differentiate between two variants of species barcoding gaps, depending on whether specimen identification or species discovery is the end goal: `local' and `global' respectively. A `local' DNA barcode gap can be applied whenever an individual specimen of a particular species is closer in distance to another member of the same species; whereas, a `global' barcoding gap is applicable whenever a threshold can be identified that separates all species \cite{collins2013seven}. While each of these hold great importance for the identification of unknown specimens, the absence of a sufficiently wide `global' barcode gap that readily distinguishes higher-level taxonomic diversity (\textit{e.g.}, phyla) does not immediately rule out the existence and usefulness of `local' gaps at lower levels of taxonomic organization (\textit{i.e.}, genus, species) \cite{koroiva2018estimating, kvist2017does}.  



Many studies, particularly early ones (see below for some taxon-specific examples), use a barcoding gap approach as a reliable species, genus, or higher-level (\textit{i.e.}, order, family) separation criterion with little discussion as to its overall utility \cite{candek2015dna}. However, both the existence and application of the DNA barcode gap are equally important: taken together, they reconcile both morphological identifications established through Linnean taxonomy with molecular identifications based on DNA sequence variation. Recognizing this, Collins and Cruickshank \cite{collins2013seven} discussed the ``inappropriate use of fixed distance thresholds" and ``incorrectly interpreting the barcoding gap". Despite strong support early-on in the DNA barcoding enterprise for the existence of the DNA barcode gap, subsequent studies have since gone on to suggest that the presence of a barcode gap at any taxonomic level is simply an artifact of insufficient specimen sampling across narrow geographic and morphologic space \cite{bergsten2012effect, candek2015dna, dasmahapatra2010mitochondrial}. In light of this observation, the interpretation of the barcode gap across taxa is not a straightforward task (\textit{e.g.}, marine gastropod molluscs: \cite{meyer2005dna}, butterflies: Wiemers and Fiedler \textit{et al.}~\cite{wiemers2007does}, spiders: Candek \textit{et al.}~\cite{candek2015dna}, annelids: Kvist \cite{kvist2017does}, leaf-footed bugs: Zhang \textit{et al.}~\cite{zhang2017species}, dragonflies/damselflies: Koroiva and Kvist \cite{koroiva2018estimating}). This may be due to the fact that the very definition of the DNA barcode gap has undergone refinement over the years. As a result, no one ``true" quantitative approach exists for measuring the DNA barcoding gap that can be unanimously agreed upon within the \\ barcoding community. Thus, the use of arbitrary and fixed distance cutoffs to separate out all taxa, such as the well-known 2\% heuristic \cite{hebert2003biological, hebert2003barcoding} and the 10$\times$ rule \cite{hebert2004identification} no longer holds \cite{collins2014known, zhang2017species}. Rather, because barcoding data is in a state of continual flux, as more specimens are collected and as taxonomic revisions are made, taxon distance thresholds should instead be directly computed from specimen DNA sequences when possible \cite{collins2013seven, young2017barcode}. 


Numerous computational and statistical methods have been proposed over the years to better quantify the magnitude of the barcode gap. For instance, the 10$\times$ rule, where interspecific variation is 10 times greater than intraspecific variation, goes some way into accomplishing this, but it cannot be readily applied to distinguish among all taxa due to differences in both taxon evolutionary and life histories. The absence of a DNA barcoding gap to reliably discriminate species can be attributed to three primary factors: (1) the recent/rapid splitting of species from the Most Recent Common Ancestor (MRCA), leading to the retention of ancestral polymorphisms as a result of incomplete lineage sorting, \\ introgressive hybridization or species synonymy; (2) the likely presence of cryptic species diversity due to lack of fixed morphological differences among closely-related taxa; and, (3) human-mediated errors (\textit{e.g.}, overlumping/oversplitting of taxa) in the identification of specimens by experts \cite{hubert2015dna, koroiva2018estimating}. Thus, the employment of taxon-specific distance \\ thresholds, as opposed to generic cutoffs, seems more reasonable. As a consequence, the adoption of a number of query-based criteria designed to aid in the reliable separation of intraspecific and interspecific distances has propagated throughout the DNA barcoding community and literature over the years. These include for instance the Best Close Match criterion employed within the TaxonDNA software \cite{meier2006dna}, and methods available in the {\tt spider} \cite{brown2012spider}, the {\tt adhoc} \cite{sonet2013adhoc} and the {\tt BarcodingR} \cite{zhang2016barcoding} R packages. In spite of the introduction of various methods to aid the solving of the species genetic separation problem, a significant knowledge gap persists: the apparent dearth of statistical \\ thoroughness in the computation of genetic distances that accompanies the majority of published DNA barcoding studies. 



\section{A Need to Improve and Maintain Statistical Rigor in DNA Barcoding Studies}

Here, evidence is presented that has greatly hampered the acknowledged and untapped potential of DNA-based specimen identification and species delineation: the lack of \\ statistical rigor in DNA barcoding. Despite having been pointed out as a clear limitation multiple times in various capacities by authors of previous studies \cite{luo2015simulation, matz2005likelihood, nielsen2006statistical, phillips2019incomplete}, the issue of statistical rigor within the context of DNA barcoding has not yet garnered the scrutiny it desperately deserves.  This may be due to the fact that there is no one set definition for statistical rigor in the literature, partly because, like science, statistics is rooted deeply in epistemology, and more generally in philosophy \cite{lindley2000philosophy}. The problem faced here however is that the majority of researchers, particularly those in life science fields, lack an appropriate level of statistical knowledge necessary for the proper application of statistical methods \cite{fieberg2020resampling}. As a result, misuse, abuse and misinterpretation of quantitative results is rampant in academic settings. By no means are DNA barcoding studies immune to this. Such na{\"i}vet{\'e} has led to the overuse of `basic' parametric statistical procedures such as \textit{t}-tests and the drawing of incorrect conclusions from \textit{p}-values \cite{wasserstein2019moving}. These and other statistical ``sins" are so widespread in academic publications that some statisticians have devoted much of their time, and even their entire careers, to writing about the most common errors made by non-statisicians and steps to take to avoid making them (\textit{e.g.}, Good \cite{good2003common}). Thus, here, statistical rigor is informally and simply defined as the use of appropriate quantitative methods (\textit{e.g.}, genetic distances) to test and justify hypotheses in light of empirical evidence (\textit{i.e.}, DNA sequence data) and uncertainty. This definition is adopted herein. Notably, it is stressed that the ubiquitous barcoding gap should be better defined on a statistical level, contingent on its use for the task of identifying unknown specimens or in describing novel species. However, as most DNA ``barcoders" are not also statisticians, the lack of a statistically-precise static definition for the DNA barcode gap is understandable, albeit one that is absolutely necessary. Although there is much that could be elaborated on here, in this brief investigation, focus is specifically placed within the context of the need for the sound interpretation of the DNA barcoding gap as a necessary and sufficient criterion to assess the overall performance of DNA barcoding in animal taxa.



In the following subsections, problems with barcode gap interpretation from the \\ standpoints of (1) requiring higher intraspecific specimen sample sizes to adequately \\ capture standing genetic variation, (2) needing better descriptive statistics, along with visualization methods, to concisely and accurately summarize taxon genetic sequence \\ distance data, and (3) necessitating more appropriate statistical inference procedures to draw meaningful conclusions from limited DNA sequence data are outlined. Throughout the present work, Meier \textit{et al.}'s~\cite{meier2008use} version of the DNA barcoding gap is employed; however, one can easily replace `maximum' with `mean' everywhere in the context of interspecific distances with the understanding that discrepancies as to which species show a barcode gap may (and often do) result. Viable solutions are then proposed which better harmonize the seemingly disparate disciplines of DNA barcoding and statistics. Moreover, the methods proposed herein also extend to the notion of (statistical) \textit{reproducibility}. Many scientific studies lack sufficient information (including detailed explanations, quantitative data and metadata) necessary to replicate original experiments. A prime example where a sufficient level of detail is crucial to convey to researchers is in the description of \\ agent-based models (ABMs), which are used widely in ecology. Typically, ABMs \\ incorporate numerous assumptions needed to establish baseline individual- and group-level behaviour in ``perfect-world" scenarios. To ensure that such rigor is not compromised, Grimm \textit{et al.} \cite{grimm2006standard} introduced and outlined a standard protocol that sought to bridge \\ challenges of ABM: \textbf{O}verview, \textbf{D}esign concepts and \textbf{D}etails (ODD). This work has since been expanded upon to more fully encapsulate the elements needed to adequately describe ABMs in a complete but succinct manner \cite{grimm2010odd}. The approaches outlined and examined below go some way into better enabling reproducibility, much like the ODD Protocol, as they are not only planted firmly in solid statistical theory, but are also straightforward to implement and easy to understand by the statistical nonexpert. While it is recognized that much of the discussion outlined herein on the absence of statistical rigor in DNA barcoding in the context of the barcode gap is highly animal-centric, potential solutions to ameliorate this problem are easily extended to other taxon groups, in particular plants and fungi, and their proposed barcode loci. Focus on animals was decided on simply because use of DNA barcoding in this group is much more straightforward and less controversial in comparison to non-animal species.



\section{Case Study: DNA Barcoding of Pacific Canada's \\ Fishes}

From this point onward, statistical approaches to better characterizing the DNA barcode gap will be framed in the context of the barcoding of Canadian Pacific fishes as a focal case study. Many fish species native to the Pacific (\textit{e.g.} Sockeye salmon (\textit{Oncorhynchus nerka})) hold strong socioeconomic and conservation importance globally, particularly as central food commodities within the supply chain. As such, in recent decades, much work has gone towards better understanding patterns of standing genetic diversity in this group to aid recovery of declining fish stocks.   



DNA barcodes were downloaded from BOLD on December 1, 2020. Specifically, \\ sequence data were taken from Steinke \textit{et al.}~\cite{steinke2009dna} (BOLD Project: TZFPC Fishes of Pacific Canada Part I) and consist of 1219 specimens representing 198 species (as of the date of download). At the time of project release and publication of Steinke \textit{et al.}~\cite{steinke2009dna}, data comprised 1225 specimens records from 201 species. Within the current dataset two specimen records (Process IDs: TZFPA062-06 and TZFPB406-05) were flagged as problematic (\textit{i.e.} misidentified) in BOLD and an additional sequence \\ (Process ID: TZFP069-04) was outside the barcode region length necessary for BARCODE compliance (\textit{i.e}, said sequence was shorter than 500 bp; \cite{hanner2009data}). Only the latter record was excluded from further analysis, leaving a sequence count of 1218. The two misidentified specimens were identified at the time of record submission to the species level as \\ \textit{Arctozenus risso} (Spotted barracudina) and \textit{Lipolagus ochotensis} (Eared blacksmelt), \\ respectively. A single record was associated with only a genus name (TZFPA198-07 \textit{Icelinus} (sculpins)), so it was removed from subsequent calculations/inferences. Further, a total of 46 singleton species were identified. These records were also excluded from downstream consideration since DNA barcode gap analysis requires the inclusion of at least two specimens per species to be meaningful. This clearly highlights the lack of adequate sufficient sampling necessary to capture standing levels of within-species genetic variation existing within many DNA barcode libraries. Included in the gap analysis were five records comprising interim species (\textit{Paraliparis} sp. (snailfishes)). Thus, only 1171 of the 1218 specimens (representing 152 species) were deemed useable. Sequence alignment necessary for determination of the DNA barcoding gap was carried out directly using the built-in amino acid-based Hidden Markov Model (HMM) aligner due to dataset size. The default Kimura-2-Parameter DNA substitution model was maintained, along with the default Pairwise Deletion option for ambiguous base and gap handling. 



Using the Barcode Gap Analysis tool available through the BOLD Workbench, results \\ revealed a total of 29 species (19.1\%) had nearest neighbour distances less than the 2\% threshold. The species \textit{L. ochotensis} showed a maximum intraspecific distance of 1.24\% and a minimum interspecific distance of 13.43\% (nearest species: Northern smoothtongue (\textit{Leuroglossus schmidti}); nearest neghbour: TZFPA187-07), while distances for \textit{A. risso}, a singleton, were 0\% and 17.72\% (nearest species: Northern pearleye (\textit{Benthalbella dentata}); nearest neighbour: TZFPB335-05), respectively. Although observed magnitudes of genetic distances for both of these species suggest that DNA barcoding ``works" and is an effective tool when it comes to specimen identification, it is nevertheless unsettling that all \\ mentioned species' nearest neighbours fall into separate genera. While this finding suggests a lack of overall specimen sampling depth for these species and perhaps Pacific fishes in general \cite{steinke2009dna}, it must further be emphasized that many Pacific species occur in deep, cold-water environments; thus, deep barcode sequence divergences are not a rare \\ phenomenon. One species, \textit{Paraliparis pectoralis}, had a maximum within-species distance of 2.27\% (minimum interspecfic distance: 9.17\%, nearest species: \textit{Paraliparis paucidens}, nearest neighbour: TZFPA048-06). Another species, the Deepwater bristlemouth \\ (\textit{Cyclothone atraria}), displayed a maximum intraspecific distance of 9.22\% (nearest \\ neighbour distance: 22.78\%; nearest species: Stout blacksmelt (\textit{Pseudobathylagus milleri}); nearest neighbour: TZFPB400-05). These two cases of high intraspecific distance is strong indication of \textit{potential} cryptic species diversity. All other maximum within-species \\ distances were below 2\%. It should be noted here that specimens assigned as \\ correctly-identifed or misidentified to a given species, as well as those individuals \\ displaying cryptic genetic variation or evidence of barcode sharing may in fact not bear these characteristics. Calculated maximum intraspecies genetic distances for \textit{C. atraria} and \textit{P. pectoralis} were based on sample sizes of only nine and 12 individuals respectively. Even though specimen sample size information is available for all species assessed by Steinke \textit{et al.}~\cite{steinke2009dna}, it is still impossible to directly discern how reliable reported genetic distance measures are, and therefore the trustworthiness of estimated DNA barcode gaps.



\section{Highlighting the Lack of Statistical Rigor in DNA \\ Barcoding}

Prior to delving any further into the primary elements that constitute the lack of \\ statistical rigor in DNA barcoding, along with the discussion of simple solutions to help aid its mitigation, some readers may have noted thus far the use of the term ``distance" to describe both genetic variability within species as well as among species. This is no mistake. These terms are synonymous from a lexical point of view, and can be (and often are) used interchangeably within general writing. However in scientific writing, this constitutes a major \textit{faux-pas}. The term ``divergence" is used in the phylogenetics literature to express differences in either number of mutations or amount of time separating taxa (\textit{e.g.}, species differing by 2\% per million years) accomplished through molecular clock measurements. Recently, DeSalle and Goldstein \cite{desalle2019review} reiterated the importance of carefully balancing word meaning and word choice in barcoding papers so that author(s)' overall intent is not blurred. Numerous highly-cited past DNA barcoding studies employing barcode gap analyses have unknowingly used the term ``divergence" to denote gene \\ variation seen across species. Even some authors of the current work are guilty of this. Such word usage bears similarity to the confusion between the terms ``species identification" and ``specimen identification", as raised by Collins and Cruickshank \cite{collins2013seven} as the third of seven deadly sins of DNA barcoding. There is an important mathematical/statistical distinction between distance and divergence which must be stressed: distances are \textit{symmetric}, whereas divergences are not. Considering two different specimens (or species), calling them $A$ and $B$, then the distance between $A$ and $B$ is equal to the distance between $B$ and $A$. Such a pattern is easily observed from examining a pairwise distance matrix of intraspecific and interspecific genetic distances. Values are identical (zero) with respect to the main diagonal (moving top left to bottom right), as a specimen or species will display zero distance from itself to itself. This can also be seen through exchanging matrix rows for columns and \textit{vice versa}. On the other hand, the notion of divergence (\textit{e.g.,} the Kullback-Leibler divergence) speaks to how different two probability distributions are from one another. Thus the term ``distance'' is employed everywhere throughout the current work when referring to intraspecific and interspecific differences. While this confusion does not directly contribute a lack of statistical rigor \textit{per se}, to this end, all future DNA barcoding studies employing barcode gap analyses should use the term ``distance" over ``divergence" to avoid any \\ potential ambiguity and confusion.  



\subsection{Improper Allocation of Specimen Sampling Effort}

Current specimen sampling efforts for DNA barcoding have been improperly \\ delegated to further the growth of public reference sequence databases such as BOLD and GenBank. Both geographic and taxonomic barcoding projects and campaigns have been far too focused on exhaustively sampling as many species as possible \cite{phillips2019incomplete, phillips2015exploration}. This assertion is immediately evident from examining BOLD species lists, where an \\ overwhelming majority of species are singletons or doubletons. For instance, while \\ specimen records are abundant for taxa like fishes and insects, they are highly lacking for birds, most mammals and especially herpetofauna (reptiles and amphibians). From this observation, it is clear that the sampling of intraspecific rather than interspecific genetic variation has been severely limited. While both extremes of genetic variation are necessary to fully comprehend and assess the scope and magnitude of species limits, taxon rarity combined with narrow sampling are typical in DNA barcoding studies \cite{ahrens2016rarity}. Barcoding initiatives should therefore instead be focused on the dense sampling of an \textit{optimal} number of \textit{specimens} per species, which should be strongly calibrated by factors such as research budget, cost and funding \cite{cameron2006will, stein2014is}. Much of the need for comprehensive sampling of intraspecific genetic variation and the appropriate magnitude of required specimen sample sizes also stems from observed phylogeographic patterns in wide-ranging taxa 
and/or the continual hybridization between evolutionarily-related species in tandem with introgression of mitochondrial genomes. Thus, many factors are at play and need to be accounted for in the determination of adequate sample sizes, both for DNA barcoding and unbiased estimation of the DNA barcode gap.
 
 
 
In the early days of the DNA barcoding endeavour, it was decided by the Consortium for the Barcode of Life (CBOL) that at least 5-10 specimens per species be collected from wide geographic regions for assembly of reference sequence libraries; indeed, this heuristic has been globally adopted by barcoding campaigns such as the Fish Barcode of Life (FISHBOL) \cite{ward2009campaign} in an attempt to limit project costs and maximize returns. However, while collection of only a few individuals of every species is a good starting point, recent studies have highlighted that such small sample sizes are likely far from adequate to capture the majority of standing haplotype variation found within species; instead, hundreds to thousands of individuals may be needed based on both empirical findings and simulation studies \cite{zhang2010estimating, phillips2015exploration, phillips2019incomplete}. Further, it is imperative that, in addition to target species, sister species also be adequately sampled. This is necessary for both the strong detection and the correct interpretation of the DNA barcode gap. In the case of monotypic genera or rare/endangered taxa, representatives from the closest allied genus should also be targeted since only a few exemplars/individuals can often be retrieved. 



\subsubsection{Estimating Intraspecific Specimen Sample Sizes with the R Package {\tt HACSim}}

The lack of a comprehensive and robust sampling of within-taxon genetic variation is a very real problem for molecular species diagnosis because it impedes the ability of DNA barcode researchers to acquire a full understanding of standing levels of intra-taxon haplotype diversity that is required to quantify the barcode gap.



To this end, the R package {\tt HACSim} \cite{phillips2020hacsim} can aid biodiversity researchers and \\ regulatory scientists in assessing current levels of specimen sampling effort reflected in genomic sequence libraries like those housed in BOLD (\textit{i.e.}, through computing the \\ observed fraction of haplotype diversity that has likely been sampled within species). The method can further assist researchers in obtaining optimal specimen sample sizes likely required to adequately capture the majority of haplotype diversity found within presumably panmictic species randomly sampled across their entire geographic/ecologic ranges. This is done through extrapolating haplotype accumulation curves and observing the point on the \textit{x}-axis where curves begin to saturate toward an asymptote. It is well known that most species within diverse taxonomic groups (\textit{e.g.}, freshwater fishes) exhibit high degrees of population structure and geographic isolation. Thus, the likelihood of observing a true species' barcode gap is increased when specimen sampling effort is high across its geographic range. Furthermore, the employment of {\tt HACSim} to better gauge required sampling depths within species means that less reliance will ultimately be placed on \\ arbitrary distance thresholds such as the 1\% cutoff employed within BOLD \cite{ratnasingham2007bold, ratnasingham2013dna} to assign Linnean names to user-submitted query sequences based on expertly-verified references. Since it has long been recognized that a given taxonomic level is not equivalent, in terms of genetic distance, across different evolutionary lineages (\textit{e.g.}, a family of insects is not equal to a family of fishes), it is reasonable to expect that species falling on separate branches of the Animal Tree of Life will warrant the use of different distance thresholds when it comes to specimen identification. In fact, it seems reasonable that the output of {\tt HACSim} can be employed to calculate optimal distance thresholds for reliable species separation. This is because, with larger specimen sample sizes and increasing spatial scale, intraspecific genetic distances will tend to increase, while distances observed among species will shrink \cite{bergsten2012effect, meyer2005dna}.



{\tt HACSim} is specifically relevant to assessing genetic variation derived from \\ Sanger-based amplicon reads obtained from any taxon under study and any molecular marker of interest. Preliminary simulation studies demonstrate that {\tt HACSim} reliably \\ suggests specimen sample sizes necessary to recover wide-ranging levels of within-species haplotype diversity (see \textbf {Chapter 4}). Thus, such a method should be of invaluable use to the DNA barcoding community-at-large. However, as we progress deeper into the realm of big data, the potential of {\tt HACSim} to aid in the characterization of next-generation sequencing (NGS) and High-Throughput Sequencing (HTS) data for environmental DNA (eDNA) and metabarcoding applications becomes clear. This said, it is critical that the capabilities of {\tt HACSim} be expanded upon, especially the ability to handle multiple specimen reads. Thus, all computational DNA barcoders should consider contributing to this endeavour.



\subsection{Failing to Properly Visualize Intraspecific and Interspecific \\ Genetic Distances}

A large majority of published DNA barcoding studies infer the detection (presence or absence) of a species' barcode gap through visualization of specimen pairwise sequence distances as either histograms or dotplots \cite{collins2013seven}. Collins and Cruickshank \cite{collins2013seven} were correct to suggest the employment of dotplots as opposed to frequency histograms to better depict the estimated distribution of species' interspecific and intraspecific distances, but they failed to offer a more thorough quantitative treatment as to why this is the case. 



\subsubsection{Circumventing the Problem with Histograms and Dotplots for Barcode Gap Display}

Histograms partition numerical data into \textit{discrete} class intervals called bins (not to be confused with BINs (Barcode Index Numbers) \cite{ratnasingham2013dna}) to more easily visualize how sample data is distributed. However, the use of histograms, while both ubiquitous as a statistical summarization method and widely-understood by many, can often muddy the true shape of probability distributions if both the bin width and number of bins in which to group data are not chosen wisely. Histograms with narrow bins tend to be more precise when density of the sample data is low; whereas, when density of observations is high, wider bin widths should be preferred because of the tendency to better expose true data signal relative to noise \cite{scott1979optimal}. Whenever bin widths are chosen to be equal in size, the \textit{height} of resulting histogram bars is proportional to the number of samples contained in each bin. Conversely, for the case of unequal bin widths, the \textit{area} of bars scales with the number of observations. Despite the added benefit of experimenting with bin widths to reveal hidden structure within data, most software now routinely employed to construct histograms, such as R's {\tt graphics} \cite{r2018language} and {\tt ggplot2} packages, utilize equal bin widths in generating histograms by default. Similarly, too small a choice of the number of bins and the histogram will be very rugged (\textit{i.e.}, have high bias); too large the number of bins and the histogram will be oversmoothed (\textit{i.e.}, possess high variance) \cite{scott1979optimal}. If DNA barcode researchers choose to continue to use equal histogram bin widths to display the barcode gap, then consideration of the optimal number of bins to employ needs to be carefully investigated. Several measures of appropriate bin numbers to use have been proposed in the statistical literature such as the robust Freedman-Diaconis rule \cite{freedman1981on}, which makes use of the sample interquartile range (IQR), or Scott's Normal reference rule \cite{scott1979optimal}, which employs the estimated (sample) standard deviation calculated from Normal distributions. Unfortunately, most heuristics (including the ones mentioned here) place a strong dependence on sample size. For instance, Microsoft{\textregistered} Excel sets the number of histogram bins to be equal to the square root of the number of data observations, whereas {\tt graphics} employs Sturges rule \cite{sturges1926choice}, basing the number of bins to scale proportionally with the base-two logarithm of the number of samples, while {\tt ggplot2} defaults to using 30 bins regardless of dataset size. For Sturges rule, bin width is computed from dividing the sample range of the data by the optimal bin number. The validity of Sturges rule in particular has been called into question as it tends to oversmooth data in the case of large samples \cite{hyndman1995problem} and there have been calls for the usage of more reliable methods. The main problem with equal bin widths is that important trends in the data may be confined to only one or a few bins. A further point worth mentioning here is that many programs (R included) default to using histogram \textit{frequencies} (counts per bin). However, this may not be ideal. Plotting based on \textit{densities} instead has the advantage of ensuring the area under the histogram is equal to one. Therefore, allowing histogram bins to vary in width for either (or both) genetic distances within or among species may be worth exploring. The reason behind employing such an approach is that it can account for bins with low numbers of observations. While having an approximately equal number of data points per bin may be ideal, such an approach is not typically seen in practice. Further, if this route is taken, alternatives for appropriate bin numbers apart from the methods outlined previously need to be examined. Because studies potentially employ different software for histogram generation, results are no longer directly comparable; thus, care must be exercised when making generalizations.



A much better alternative to displaying the DNA barcoding gap is to rely on the \\ \textit{continuous} variant of the histogram, kernel density estimation (KDE) plots \cite{rosenblatt1956remarks, parzen1962estimation}, to more accurately inform on the actual population distribution of the barcoding gap through depiction of intraspecific and interspecific pairwise genetic distances as smooth curves. KDE works by weighting data observations relative to their distance to other \\ similar-magnitude data points. Much like histograms however, KDE often requires careful parameter selection, in particular regarding the kernel type and the kernel bandwidth. The kernel type strongly defines the overall shape that the density curve takes on, whereas the kernel density bandwidth controls the amount of smoothness of the generated curve. Optimal choice of these parameters is crucial so as to not distort real patterns present within the data. Most modern software (such as R) employ defaults which tend to work well under a wide variety of situations, but also give the user fine control over parameter initialization. However, automatic settings can sometimes lead to undesirable results. R for instance employs a Gaussian kernel and chooses the kernel bandwidth to be equal to the standard deviation of the kernel itself; this should be sufficient as far as estimation of the DNA barcode gap is concerned. Often with kernel density estimation, data may extend beyond those observed from histograms. In particular, data that are constrained to only positive support values can end up having negative density values, which for genetic distances, is not biologically meaningful. In practice however, this is not an immediate concern since truncation methods exist to ensure that data located at the boundaries of KDE plots have positive support.    



The dotplot approach to inferring the barcode gap (\textbf{Figure 5.1}) is simple: on a plot of maximum intraspecific genetic distances (displayed on the $x$-axis) \textit{versus} minimum \\ interspecific distances (shown on the $y$-axis), represented by points for every barcoded species, a line corresponding to the function $y=x$ is drawn. Points occurring above this line suggest that a barcode gap is present for a given species and that DNA barcoding ``works". In contrast, points falling below the 1:1 line for any species suggest that the DNA barcode gap is absent, and thus barcoding fails to tell specimens apart. Often, points plotted in this fashion overlap tightly, making species-by-species visual inspections difficult. \\ \textbf{Figure 5.1} clearly shows that many Canadian Pacific fish species exhibit maximum 
\\ intraspecific distances very close to, or equal to, zero. This strongly indicates that adequate specimen sampling needed to characterize standing haplotype diversity at the species level is severely lacking.

\begin{figure}[H]

\centering

\includegraphics[width=0.8\textwidth]{"Figure 1".pdf}

\caption[Depiction of the DNA barcode gap as a traditional dotpot for Canadian Pacific fishes assessed by Steinke \textit{et al.}~ \cite{steinke2009dna}.]{Traditional dotplot for visualizing the DNA barcode gap for a range of Canadian Pacific fishes assessed by Steinke \textit{et al.}~\cite{steinke2009dna} and generated using the {\tt ggplot2} \cite{wickham2016ggplot2} R package. Data come from the BOLD Workbench's Barcode Gap Analysis tool and comprise 1171 specimens from 152 species (\textit{c.} 7.7 specimens per species on average). Points lying above the 45$^{\circ}$ line indicate that species show a barcode gap and are readily identified via molecular barcodes. On the other hand, points falling below the 1:1 line suggest that species lack a barcoding gap and thus are not easily diagnosed through their DNA barcodes. Most species assessed here (146 of 152 species (96.1\%)) display a barcode gap since the minimum interspecific genetic distance exceeds the maximum intraspecific genetic distance. Despite this, evidence of species showing a barcode gap may in fact be an artifact of limited sampling of within-species haplotype variation. The species \textit{Cyclothone atraria} at the point (9.22, 22.78) is clearly visible as an extreme outlier and signals possible cryptic species variation.}
\end{figure}



The use of traditional dotplots to display the barcode gap would be better represented as half-logarithm dotplots (\textbf{Figure 5.2}) which plot sorted log-transformed genetic distances for every species included in a taxon dataset against the number of species sampled \cite{steinke2009dna, steinke2009barcoding}. A horizontal line is then drawn at the 1\% mark (or similar threshold), allowing for good separation of intraspecific distances from nearest neighbour distances. Plotting sorted genetic distances in this manner allows for relative differences to be easily seen among species \cite{steinke2009dna}. Further, through employing a log transformation of species' genetic distance data, interesting patterns are more easily spotted without worry for any loss of information. This is the case for two reasons. First, since $y = \log_{a}(x)$ (where $a \in (0, 1) \cap (1, \infty)$ and $x>0$) is \textit{monotone}, the order of plotted points is preserved. Second, the log transform is \textit{variance-stabilizing} because it has the effect of making positively-skewed data less skewed through removing any dependence existing between the mean and variance of a set of data observations. Without such a transformation in place, sample observations would likely display varying levels of heteroscedasticity (\textit{i.e.}, non-constant variance). Similar to the $y$-axis of \textbf{Figure 5.1}, numerous data points (representing over 60 fish species) lie directly on the $x$-axis, indicating a complete lack of sufficient specimen sampling (\textbf{Figure 5.2}). Despite its promise, it appears that the modified dotplot has not caught on within the DNA barcoding community outside a select few fish DNA barcoding studies \cite{steinke2009dna, steinke2009barcoding}. 

\begin{figure}[H]

\centering

\includegraphics[width=0.8\textwidth]{"Figure 2".pdf}

\caption[Depiction of intraspecific and interspecific genetic distances as a modified dotplot for Canadian Pacific fishes assessed by Steinke \textit{et al.}~\cite{steinke2009dna}.]{Half logarithm dotplot for the display of species' genetic distances modified from Steinke \textit{et al.}~\cite{steinke2009dna} for fishes from Pacific Canada. Plotted data comprise those specimens originally analyzed by Steinke \textit{et al.} \cite{steinke2009dna} (\textit{i.e.}, 1225 specimens from 201 species). Most sampled species are resolved at the 1\% log level of genetic distance.}

\end{figure}



A much more intuitive means of displaying intraspecific distances and interspecific \\ distances is through using ``quadrant plots" (\textbf{Figure 5.3}) because they can be employed to directly detect ``outlier" and problematic species in need of closer examination. In this approach, as in the generation of traditional and half-logarithm dotplots, barcoded species are depicted as points on a plot of maximum intraspecific distances on the $x$-axis \textit{versus} minimum interspecific distances on the $y$-axis. Points fall into one of four categories in positive Cartesian space, depending on a predefined species distance cutoff (2\% typically). Moving in a clockwise fashion from the top left corner, each category can be viewed as a case of either ``barcoding success" or ``barcoding failure". Quadrant I corresponds to the case where species are easily discriminated using DNA barcoding and reflects concordance with currently accepted Linnean taxonomy (\textit{i.e.}, interspecific distances are greater than the prespecified level cutoff, while intraspecific distances are less than the chosen threshold --- a ``success"). Species falling into Quadrant II likely represent cryptic complexes (\textit{i.e.}, both intraspecific distances and interspecific distances are greater than the prespecified level cutoff --- a ``failure"). Species in this partition are indistinguishable through morphology alone and as a result are lumped under a single species name by taxonomists. Quadrant III encompasses evolutionarily young species that have recently diverged from the MRCA (i.e., not enough time has elapsed to allow nucleotide differences in the barcode region to accumulate --- a ``failure"). This category can also include species that are known by various synonyms. Finally, Quadrant IV includes likely misidentified specimens or instances of hybridization between closely-related species --- a ``failure"). Of all the case study species, \textit{C. atraria} is the only species that would fall into Quadrant II. Based on computed genetic distances, both \textit{Arctozenus risso} and \textit{Lipolagus ochotensis} would be classified as belonging to Quadrant I; yet BOLD categorizes each of them as misidentified. This result is telling: it strongly suggests that \textit{Lipolagus ochotensis} was represented in Steinke \textit{et al.}'s~\cite{steinke2009dna} dataset by only a handful of collected specimens. Indeed, this is the case with only four sampled specimens. Thus, the plausibility of both \textit{Arctozenus risso} and \textit{Lipolagus ochotensis} as barcoding ``successes" (and therefore presenting a real barcode gap) should be immediately called into question. Like the half-logarithm dotplot, the quadrant plot approach has seen very limited use in barcoding studies, despite its inherent simplicity. Such plots appear to have only been employed in two previous publications \cite{hebert2004identification, hubert2015dna}. 

\begin{figure}[H]

\centering

\includegraphics[width=0.8\textwidth]{"Figure 3".pdf}

\caption[Depiction of species' genetic distances as an altered quadrant plot taken from Hubert and Hanner \cite{hubert2015dna}.]
{Quadrant plot for the depiction of species' genetic distances reproduced with modification from Hubert and Hanner \cite{hubert2015dna}. Such plots are informative since problematic and/or ``outlier" species can be easily detected and the success/performance of DNA \\ barcoding assessed. Species are partitioned into four mutually exclusive groups (labelled I-IV) based on observed magnitudes of intraspecific and interspecific genetic distances. Here, a 2\% distance threshold is assumed to separate most taxa. Solid black lines aid visual resolution of the quadrants. The blue dot at the approximate point (9.22, 22.78) within Quadrant II represents \textit{Cyclothone atraria}, a likely cryptic species complex. Plot axes show the relationship to the density curves shown in Meyer and Paulay \cite{meyer2005dna}.  Theoretically, between-species genetic variation should greatly exceed barcode sequence variation observed within species (Quadrant I; minimum interspecific distance $-$ maximum intraspecific distance $>$ 2\%). Practically, this will only be the case when specimens have been adequately sampled.}

\end{figure}


Whereas the abovementioned visual tools offer strong proof-of-concept of the DNA \\ barcode gap, one element that they fail to reveal, however, is whether a barcoding gap likely \textit{exists}. To properly address this question, more rigorous statistical methods are required.


\subsection{Inconsistent, Inappropriate Use, or Absence of Inferential \\ Statistical Procedures in DNA Barcoding}

Attempts to place DNA barcoding on more statistically-solid ground have been \\ undertaken several times before, particularly with regard to specimen classification (\textit{e.g.}, Matz and Nielsen \cite{matz2005likelihood}, Nielsen and Matz \cite{nielsen2006statistical}, Abdo and Golding \cite{abdo2007step}, Austerlitz \textit{et al.}~\cite{austerlitz2009dna}, Lou and Golding \cite{lou2010assigning}, Zhang \textit{et al.}~\cite{zhang2012fuzzy}) and (single-locus) species delineation (\textit{e.g.}, Fujisawa and Barraclough \cite{fujisawa2013delimiting}, Monaghan \textit{et al.}~\cite{monaghan2009accelerated}, Pons \textit{et al.}~\cite{pons2006sequence}, Reid and Carstens \cite{reid2012phylogenetic}, Puillandre \textit{et al.}~\cite{puillandre2011abgd}, Zhang \textit{et al.}~\cite{zhang2013general}). Many of these proposed methods have seen widespread usage, while others seem to be rarely employed in certain instances due to their inherent mathematical complexity and/or black-box nature. Here, the intent is to highlight the increased need for more rigorous statistical procedures for better characterizing the DNA barcode gap by pointing to various efforts that have been made in merging statistical theory with specimen identification/classification throughout the years.

 

Perhaps the first instance of the use of statistical algorithmic approaches in DNA \\ barcoding was for the purpose of specimen classification. Such methods relied mostly on ideas from classical inferential paradigms such as likelihood theory and subjectivist (Bayesian) thinking, whereas others took inspiration from more modern models, \\ particularly machine learning. One promising, yet grossly undervalued technique worth mentioning here is the probability of correct identification (PCI) \cite{martin2020figures, spouge2012practical}. While the PCI has many variants, its primary function is to serve as a simple metric of DNA barcoding efficacy given a richly-populated and fine-tuned reference database. The PCI statistic has mostly seen use around appropriate marker selection for DNA barcoding, particularly in regard to challenging taxa such as fungi, plants and protists. At its heart, the PCI is nothing more than a binomial proportion whose sampling distribution is easily estimated using resampling procedures. From here, it is trivial to calculate quantities of interest such as standard errors and confidence intervals. In essence, the strong mathematical and statistical theory that underlies the PCI is what is missing and should be emulated in future DNA barcoding studies employing the barcode gap.

 

The use of statistical approaches for species delimitation has also generated much interest. Before delving into this topic further however, it must be stressed precisely how evolutionary biologists view the notions of species and speciation. While systematicists agree on many fronts, one aspect on which there is large disagreement is in defining of species themselves. The argument for the use of DNA barcoding as a species delimitation tool necessarily rests on the adoption of a workable species concept that is congruent with one's belief system concerning what is known or expected to be true of focal taxa under study \cite{miller2007dna, pante2015species, will2005perils}. Unfortunately, deciding on a unified ``barcoding species concept" that is both adequate and universally applicable across the entire Eukaryotic Tree of Life is no simple task since such a definition would have to satisfy properties, such as reproductive isolation and monophyly, inherent in the more than two dozen already-existing species concepts found throughout the literature \cite{dequeiroz2007species}. DNA barcoding itself will not succeed in uncovering deep biological and evolutionary relationships existing among taxa \cite{will2005perils}. This is the case since multiple lines of evidence (\textit{e.g.}, morphology, ecology, evolutionary/life history, geography or behaviour) must be factored into decision-making regarding the categorization of taxa into groups reminiscent of species using an integrative taxonomic framework \cite{dayrat2005towards}. Thus, it has been strongly cautioned to tread down this road carefully, \\ considering various strategies to approaching species demarcation tasks, as well as \\ explanations for species existence, origin and formation over space and time \cite{carstens2013fail}. The majority of proposed theoretical approaches in this regard have been strongly centered on population genetic and coalescent theory \cite{kingman1982coalescent}. Despite this, methods like that of Birky \textit{et al.}'s~\cite{birky2010using} and Birky's \cite{birky2013species} $K/\theta$ ratio hold much promise in providing a deeper theoretical basis for the DNA barcode gap and a direct means of testing for its existence; however, their overall performance depends highly on the specific model of speciation assumed to characterize dynamics of taxa under examination \cite{fujisawa2013delimiting, kapli2017multirate, monaghan2009accelerated, pons2006sequence, puillandre2011abgd, zhang2013general}. For example, rapid adaptive radiation events having occurred recently in the evolution of a taxon (such as Annelida) are known to complicate both local and global barcode gap detection since allopatrically-speciating populations would show comparable nucleotide makeup in the absence of gene flow \cite{kvist2017does}. Hubert and Hanner \cite{hubert2015dna} noted a deep \\ connection between the coalescence of two distinct evolutionary lineages within a given gene tree and the observation of a barcode gap reliably separating intraspecific from \\ interspecific sequence variation. Specifically, once lineages have effectively sorted, both specimen identification and species discovery tasks become easier. 


  
In 2011, Puillandre \textit{et al.}~\cite{puillandre2011abgd} introduced the widely-popular Automatic Barcode Gap Discovery (ABGD), a nonparametric statistical method to discriminate species based on the existence of the barcode gap, using avaliable DNA sequence data, as opposed to generating taxon phylograms beforehand. Prior to this, heavy reliance fell upon the Generalized Mixed Yule Coalescent and its many variants \cite{fujisawa2013delimiting, monaghan2009accelerated, pons2006sequence, reid2012phylogenetic}. GMYC is an extremely parameter-sensitive, time- and memory-consuming model-based approach to species \\ delimitation based on branching patterns observed within ultrametric phylogenies. \\ Resulting trees are generated using third-party software such as Bayesian Evolutionary Analysis Sampling Trees (BEAST; \cite{drummond2007beast}) or MrBayes \cite{huelsenbeck2001mrbayes}, and analyzed using the \\ {\tt splits} (SPecies LImits by Threshold Statistics) R package \cite{ezard2017splits}. Since then, other methods to delimit species have been introduced to analyze barcode data (\textit{e.g.}, Poisson Tree Processes (PTP) and its relatives \cite{kapli2017multirate, zhang2013general}, Haplowebs \cite{dellicour2015delimiting}, ASAP (Assemble Species by Automatic Partitioning) \cite{puillandre2021asap}) more efficiently. Earlier approaches such as haplotype parsimony networks \cite{templeton1992cladistic}, constructed using software like TCS \cite{clement2000tcs}, have found their way into DNA barcoding, despite known interpretational issues such as the tendency to form disconnected subnetworks, or the inclination to group species together within the same node \cite{hart2007things, phillips2019incomplete}. In addition, the default 95\% detection limit (\textit{i.e.}, the probability of parsimony) employed within TCS is largely arbitary; users can set this value to range anywhere from 90-99\% \cite{clement2000tcs}. Thus, the choice of distance cutoff can have a large effect on the outputted network topology. The above methods can differ greatly in the number of species delimited. Both Dellicour \textit{et al.}~\cite{dellicour2018hitchhiker} and Luo \textit{et al.}~\cite{luo2018comparison} note that GMYC tends to overestimate (oversplit) species, whereas underestimation of species (\textit{i.e.}, undersplitting) is evident for ABGD. The BIN approach \cite{ratnasingham2013dna} seems to be a good compromise to its predecessors as it is fast to run, straightforward to implement and resulting output is easily interpreted \cite{kekkonen2014dna}. A BIN comprises a unique alphanumeric code corresponding to a tight cluster of closely-related haplotypes. The BIN framework employs hierarchical clustering (via the REfined Single Linkage (RESL) algorithm), along with Markov clustering, and often suggests species numbers between the extremes of ABGD and GMYC through \\ partitioning DNA sequences into four mutually exclusive groups largely reflective of actual species: MATCH, MERGE, SPLIT and MIXTURE, on the basis of genetic distances \cite{ratnasingham2013dna}. These presumptive groups or operational taxonomic units (OTUs) are biologically interpretable: MATCHES conform to established Linnean taxonomy;  MERGES indicate that distinct species are indistinguishable through DNA barcoding and should be combined under a single species name; SPLITS reflect the presence of multiple species under a common Linnean name (\textit{i.e.}, cryptic species diversity); finally, MIXTURES reveal possible specimen misidentifications or instances of introgression/hybridization \cite{ratnasingham2013dna, serrao2014calibrating}. A direct relationship exists between BIN categories and quadrant plot categories mentioned 
\\ previously: MATCHES correspond to Quadrant I; SPLITS make up Quadrant II; MERGES fall into Quadrant III and MIXTURES lie in Quadrant IV. However, despite its promise, several major drawbacks to the use of the BIN system as a suitable species proxy exist. First, it is a black box whose underlying algorithm is not well-understood by researchers outside the DNA barcoding community, such as regulatory scientists. Secondly, BINs are inherently dynamic and therefore unstable over time. As more records are added to barcode libraries, new BINs are formed and existing ones are coalesced. This behaviour mirrors that of OTUs and their gradual replacement within the metabarcoding community by Amplicon Sequence Variants/Exact Sequence Variants (ASVs/ESVs). Part of these problems may stem from the fact that the RESL algorithm remains unpublished (though aspects of the BIN framework have been successfully patented), making dataset collation and comparision difficult. Currently, there is functionality within BOLD's Workbench to cluster DNA sequences into OTUs via RESL, but no easy and direct way to compare species delineation methods using both simulated and real taxon barcode data since sequences must already reside in BOLD in some form (\textit{i.e.}, as a public or private dataset).



\subsubsection{Framing the DNA Barcode Gap as a Statistical Hypothesis}

There is a need to define the barcode gap more formally as a composite (one-sided) statistical hypothesis test. An analogy here can be made to testing the hypothesis that a gene evolves neutrally in a species population. Such hypotheses can be assessed using a wide variety of tests such as Tajima's $D$ \cite{tajima1983evolutionary}. In the present case, the null hypothesis of no barcode gap for a species would be tested against the alternative hypothesis that a barcode gap exists. Mathematically, 

$$H_0: \textrm{minimum interspecific distance $-$ maximum intraspecific distance} \leq d_0$$ $$\textit{versus}$$ $$H_1: \textrm{minimum interspecific distance $-$ maximum intraspecific distance} > d_0.$$

\noindent where $d_0$ is a predefined cutoff for species separation (say, $d_0$ = 2\%). Here, the null hypothesis ($H_0$) is assumed to be true, unless unsupported by the observed data. In this case, one fails to accept the null hypothesis in favor of the alternative. That is, it is assumed that present DNA sequence data do not support the existence of a barcode gap at the species level. Under this scheme, it is easy to distinguish between Type I errors (false positives) and Type II errors (false negatives). A false positive is analogous to taxonomic oversplitting (\textit{i.e.}, nearest neighbour distance $<$ maximum intraspecific distance); whereas, excessive lumping of species (\textit{i.e.}, nearest neighbour distance $>$ maximum intraspecific distance) strongly indicates that a false negative error has been made \cite{hubert2015dna, meyer2005dna}. A one-tailed test is chosen here, as opposed to the more widely employed two-sided test since between-species genetic variation usually exceeds that seen within species, with few exceptions. Such an approach leads to a more powerful test with greater flexibility than would be allowed using a two-sided test. 



An immediate challenge exists in formulating an appropriate hypothesis test statistic for the barcode gap. Test statistics are usually of the form

$$ T = \frac{d - d_0}{\textrm{SE}[d]} $$

\noindent where $d$ is the observed difference between minimum interspecific distance and maximum interspecific distance and SE denotes estimator standard error. For already well-sampled species (\textit{i.e.}, those with a large number of collected specimens), the above test statistic would approximately follow the standard Normal distribution whenever $H_0$ is true. \\ Unfortunately, in the case of small specimen sample sizes, deriving an expression for the standard error of the estimated barcode gap would be difficult and the distribution of said test statistic would also not be obvious.

 

Framing DNA barcoding in a statistical way is clearly needed, since for \\ densely-sampled clades, a barcode gap is almost surely to exist. Through employing deep taxon sampling schemes, DNA barcode researchers will be able to more easily detect a true species' barcode gap when one is actually present.  


\subsubsection{The Use of Nonparametric Bootstrapping to Estimate the DNA Barcode Gap}
 
In addition to simple point estimates (and associated standard errors) of the barcoding gap for varying taxa which are widely reported (\textit{e.g.}, Wiemers and Fiedler \cite{wiemers2007does}) future studies should also report confidence interval (CI) estimates around the estimated \\ population (or ``true") maximum intraspecific distance, minimum interspecific distance and the barcode gap using sample data of intra- and interspecific distances. Confidence intervals, unlike $p$-values, are more strongly favored within the statistical literature. A simple but na{\"i}ve solution in this regard is to form CIs using the data at hand; however, this requires the strong assumption that genetic distances are drawn from a large \\ normally-distributed population; in reality, the sampling distribution of pairwise distances is unknown since it is likely to be highly taxon-dependent. This should come as no surprise since genomic markers employed to assign taxon-level matches to unknown specimens using DNA barcoding show varying rates of molecular evolution both within and across taxonomic groups. These observed differences in taxon molecular evolutionary rates \\ strongly affect fundamental processes at both the microevolutionary (\textit{e.g.}, random genetic drift, mutation, natural selection) and macroevolutionary (\textit{e.g.}, speciation) scales.



Thus, a better approach to reporting parameter estimates, which does not require the sampling distribution to be known \textit{a priori}, and relaxes distributional assumptions through allowing for reasonably small sample sizes, is to employ nonparametric bootstrapping to continually resample from observed distances a large number of times (say, 10000 times) uniformly (\textit{i.e.}, with equal probability) with replacement \cite{efron1979boot}. Sampling with replacement ensures that drawn observations are both independent and identically distributed; that is, sampling a given observation has no bearing on the occurrence of a future observation and all observations are generated from the same underlying statistical population. The idea here is that, for a large number of bootstrap replicates, the distribution of resampled distances (\textit{i.e.}, the bootstrap sampling distribution) mimics the actual distance distribution for the taxon under study quite closely. Such a scheme is analogous to bootstrapping in phylogenetic inference to assess how well nodes within neighbour-joining trees support the observed data \cite{efron1996bootstrap, felsenstein1985confidence}. Because a test statistic need not be known in advance, bootstrap results can be immediately used to form appropriate level (\textit{e.g.}, 95\%) bootstrap confidence intervals for the population barcoding gap. Statistical interpretation of such constructed intervals is relatively straightforward: if the intervals contain the value $d_0$, then the \\ hypothesis that the maximum intraspecific distance does not differ significantly from the minimum interspecific distance at the hypothesized value $d_0$ cannot be rejected at the stated significance level (\textit{e.g.} $\alpha$ = 5\% for 95\% confidence). Put another way, if $d_0$ falls within the obtained CI, then the hypothesis that no barcode gap is present cannot be rejected at the chosen level of statistical significance. 



Nonparametric bootstrapping is known to perform poorly in certain situations. One such failure of the traditional bootstrap is in the estimation of extreme order statistics such as the population minimum or the population maximum. Standard bootstrapping, sometimes termed the $n$-out-of-$n$ bootstrap, works by drawing resamples of the same size as the original sample. As the ``revised'' DNA barcode gap is defined in terms of the maximum intraspecific distance and minimum interspecific distance, the usual \\ bootstrapping procedure detailed above is not applicable. It is worth mentioning that the $n$-out-of-$n$ bootstrap would indeed work as expected under the ``old" definition of the barcode gap, used prior to Meier \textit{et al.}~\cite{meier2008use}, since that definition involved only statistical means. Fortunately, there is an immediate remedy available. The trick is to take resamples of a \textit{smaller} size than the original dataset \cite{bickel1997resampling}. This technique is known as the $m$-out-of-$n$ bootstrap, where $m < n$. In employing such a method, the variability of corresponding estimates will be higher (larger) than in the regular bootstrapping procedure whereas the bias will be low \cite{chernick2007bootstrap, chernick2011bootstrap}. While this result may appear counterintuitive at first, assuming the variance of an estimator of interest is both constant and finite, said estimator's standard error will be smaller for a larger number of observations and larger for smaller sample sizes. Since $m < n$, another approach worth examining is random subsampling, which involves sampling \textit{without} replacement \cite{politis1999subsampling}. The optimal choice of $m$ however is not obvious and can have a significant impact on obtained results. Therefore, algorithms for selecting appropriate values of $m$ (such as that presented in Bickel and Sakov \cite{bickel2008choice}) must be investigated. In any case, $m$ should be chosen such that it along with $n$ both approach infinity, while at the same time ensuring that the quantity $\frac{m}{n}$ approaches zero \cite{chernick2007bootstrap, chernick2011bootstrap}. For instance, letting $m$ = $\sqrt{n}$ or $m$ = log($n$) would satisfy this condition. Regardless, the above bootstrapping approach should be used to report point estimates and desired level CIs for ``true" maximum intraspecific distances and interspecific distances, as well as population barcoding gaps in any and all future taxon-specific DNA barcoding studies (especially reference sequence library publications).



\section{Critically Evaluating the Concept of the DNA Barcode Gap in Conservation and Regulatory Contexts}

Over the years, several publications cited herein (and elsewhere) have successfully \\ harnessed and exploited the awesome power of DNA barcoding in biodiversity and \\ regulatory settings. On the flip side, numerous biodiversity-focused studies have clearly demonstrated the performance of DNA barcoding on the basis of the observed difference between intraspecific and interspecific genetic variation --- the DNA barcoding gap. What appears to be missing however are more studies lying at the intersection of these two extremes, recent examples of which include Lee \textit{et al.}~\cite{lee2019towards}. The statistical approaches put forth and explained in detail here, in addition to existing bioinformatic tools that directly compute estimated barcode gaps (\textit{i.e.}, ABGD, ASAP) can greatly aid in providing strong support for the in-depth assessment of the DNA barcoding gap as a foundationally-rigorous concept, something that is nonexistent within most studies bearing a socioeconomic flavour.



Below, some specific use cases of DNA barcoding as they relate to characterizing both global and local barcode gaps in conservation and regulatory contexts are highlighted. Language is explicitly borrowed from Collins and Cruickshank \cite{collins2014known}, who categorize taxon sequences into four independent groups: ``known knowns", ``known unknowns", ``unknown knowns" and ``unknown unknowns". ``Known knowns" are well characterized and curated species in existing barcoding libraries, whereas ``known unknowns" are formally described species, yet lack full library representation. Conversely, ``unknown knowns" reflect \\ possibly divergent/cryptic lineages among described species found in reference databases like BOLD. Lastly, ``unknown unknowns" pertain to undescribed or cryptic species without records in existing libraries. The goal here is to dispel the perceived subtleties associated with employing DNA barcodes for such applications, especially in light of reference  \\ sequence libraries reflecting an incomplete sampling of natural variation. While scenarios mentioned here do not form an exhaustive list, they nevertheless cover a broad range of possibilities necessary for adequately conceptualizing the notion of a DNA barcoding gap.



\textbf{Scenario I}: A geographic region in the Pacific has a well-known fish species \\ biodiversity. Several barcoding campaigns dedicated to monitoring fisheries bycatch over the years have resulted in a hypothetical library that is 98\% complete. All species in the library comprise more than 20 sequences. In this case, the DNA barcode gap is probably almost entirely redundant. A given specimen query will likely be identical (or differ by only a few basepairs) to one already in the library. A nearest neighbour assignment with an arbitrary 1-2\% threshold (or none at all) will correctly identify most queries to the species level, except in the case of barcode haplotype sharing. Here, only detection of the local barcode gap is of interest. Adjusting and recalculating the global DNA barcoding gap is unlikely to affect taxon classifications since observing rare species or new lineages is extremely unlikely with increased sampling intensity. Thus, the library mostly consists of ``known knowns" and ``unknown knowns". The takeaway here is that as reference library coverage increases, estimating the barcode gap becomes less relevant. Incorporation of sample size estimation tools like that of Phillips \textit{et al.}~\cite{phillips2020hacsim} will add extra reassurance that specimen sampling is sufficient.



\textbf{Scenario II}: A poorly-sampled tropical region in Costa Rica has three times as many lepidopterean species as North America. Barcoding efforts have resulted in a library that is only 60\% complete, where most species are represented by less than three sequences. Further, the taxonomy of many groups is uncertain. Due to low library coverage, there are many ``known unknowns" and ``unknown unknowns" which are likely to be encountered in real samples. Here, quantifying the barcode gap parameter space is critical to establishing a threshold by which queries can be assigned correctly as unknowns, thus limiting the rate of false positives \cite{dasmahapatra2010mitochondrial, hickerson2006dna}.



\textbf{Scenario III}: The Canadian Border Security Agency (CBSA) has intercepted a \\ shipment of fish fins from a potentially endangered shark species being trafficked for illegal sales. To guarantee a conviction, a confirmed scientific name is needed. Unfortunately, specimens cannot be matched to any publicly-available sequence records in BOLD or GenBank. Thus, the presence of ``known unknowns" in the seized samples greatly \\ complicates matters. DNA barcode gap and sample size estimation would become key pieces of evidence capable of providing statistical uncertainty of species assignment in this case.



\textbf{Scenario IV}: A research team wishes to characterize alpha diversity between habitats or environments within an unexplored region in Madagascar for the purpose of completing a biodiversity survey. The goal is to infer broad endemicity and phylogeographic trends, as opposed to individual-level diversity; however, little in way of a DNA barcode reference library currently exists to aid this effort. There are many ``known knowns", ``known \\ unknowns", ``unknown unknowns" and ``unknown unknowns". De novo species \\ delimitation tools based on phylogenetic trees, such as GMYC and PTP, as well as those approaches utilizing genetic distances, including ABGD, ASAP and RESL, are necessary to addressing problems with specific methods and breakdown of statistical model \\ assumptions. Here, local barcoding gaps are less important as no single method will be solely relied upon for any individual taxon. However, crude estimation of a global barcode gap is possible.

  

\textbf{Scenario V}: A taxonomist wishes to discover new species from samples of \\ recently-collected specimens comprising many ``known knowns", ``known unknowns", \\ ``unknown knowns" and ``unknown unknowns". As a first step, established single-locus delimitation algorithms are used to triage individuals into species-like units. Next, specific species hypotheses are tested through combining aspects of integrative taxonomy with more sophisticated Bayesian multilocus delineation approaches based on the multispecies \\ coalescent (\textit{e.g.}, *BEAST \cite{heled2010bayesian}, BPP \cite{yang2012bayesian}, Bayes factors \cite{grummer2014species}). In this case, the global barcoding gap is of no real concern, except as a rough calibration/benchmark against known members at the genus level. Multigene species demarcation approaches will be superior in this respect because speciation events will be more readily ascertained.



\textbf{Scenario VI}: It is suggested that DNA barcoding could be applied to inventory diversity in a poorly-understood family of invertebrates. Since there are only ``unknown unknowns", the global DNA barcode gap could be employed to justify that genetic and morphological characters are congruent; thus, conducting further sequencing is worthwhile. 



\section{New Avenues for Estimating the DNA Barcode Gap}

Finally, it is important to draw upon future promising avenues for continued work on accurately estimating the DNA barcoding gap. One potential application in this regard includes statistical mixture models which can account for genetic differences observed within and among species  for the purpose of molecular specimen assignment. Mixture models offer great flexibility when it comes to accomplishing this task because correlations in haplotype diversity existing at the species level can be easily incorporated into such modelling frameworks. Much effort has gone into the development of easy to use \\ computational tools to fit mixture models to a wide variety of data. A notable example in this regard that may prove valuable for barcode gap estimation is the R package {\tt mclust} \cite{scrucca2016mclust}, software that has seen widespread use for the task of parametric model-based \\ clustering in recent years.



Statistical methods for delineating species can inherently be viewed as ``mixture \\ models". All proposed species delimitation approaches to date find the optimal partition of DNA sequences into mutually-distinct groups that are highly reflective of actual species. Thus, the problem of species separation boils down to that of a simple \\ clustering/classification task. The majority of methods generate these clusters on the basis of estimated phylogenetic relationships (\textit{e.g.}, GMYC, PTP), along with an assumed \\ parametric model of species generation (\textit{e.g.}, birth-death process, Yule process), whereas others simply use the DNA sequences themselves (\textit{i.e.}, ABGD, ASAP) to arrive at a \\ plausible solution in a nonparametric fashion. In recent years, novel ``hybrid" approaches to tease out species have been published. Notably, algorithmic methods such as Fujisawa and Barraclough \cite{fujisawa2016rapid} and Jones \cite{jones2017algorithmic} stray away from objective single-locus likelihood inference to also include subjective multilocus Bayesian inferential frameworks. \\ Regardless, these and other related approaches may prove valuable in aiding better \\ detection of the DNA barcoding gap.



Another approach that should be investigated and applied to address questions about the DNA barcode gap is the employment of nearest-neighbour and other machine learning methods used in clustering and classification tasks (\textit{e.g.}, van Velzen \textit{et al.}  \cite{vanvelzen2012dna}). However, the widespread success of machine learning methods is due greatly to the availability of large amounts of training data that feed and nurture artificial intelligence (AI) algorithms, a factor that poses problems for undescribed species, rare taxa and those with narrow geographic distributions (\textit{e.g.} endemic species, monotypic taxa). With an arsenal of \\ statistical tools like mixture models and nearest-neighbour methods in hand, practitioners will be better equipped to estimate important quantities central to DNA barcoding, \\ including species separation thresholds.



Although not a statistical issue \textit{per se}, the increased need for the sequencing of multiple genetic loci, particularly nuclear genes, to solidify confidence in specimen assignment and aid resolution of taxon boundaries, cannot be stressed enough \cite{eberle2020plea}. Much like the adoption of the rbcL and matK chloroplast genes for DNA barcoding of land plants, a similar case can be made for a dual, or better yet, multiple, mitochondrial-nuclear gene system for barcoding of metazoan taxa. COI has been demonstrated to lack sufficient discriminatory power for identification in groups such as sharks and the aptly named ``problem children" (Cnidaria and Porifera) \cite{bucklin2011dna} to name a few --- all which show remarkably low rates of molecular evolution. In the case of animal DNA barcoding, several molecular regions (preferrably both mitochondrial and nuclear) should be sequenced across the same sampled specimen whenever possible; in reality however, this is rarely done. While other \\ International Barcode of Life (iBOL) member nations (\textit{e.g.}, those in Europe) have accepted the move toward multilocus DNA barcoding with open arms, it seems that Canada, \\ ironically, is not one of them. The largest global hub for DNA barcoding, the Centre for Biodiversity Genomics (CBG), perhaps appears to remain somewhat fixated on the promise of single-marker barcoding for the construction of reference sequence libraries and for the progression of biodiversity science as a whole. One can even argue that Canadian DNA barcoding's staunch position on maintaining the \textit{status quo} and its reluctance to embrace necessary change, due to the overwhelming fear of becoming irrelevant, has greatly hindered the timely transition into the vast and exciting realm of ``next-generation DNA barcoding" \cite{taylor2012emergent}. This is clearly evident from the fact that the majority of specimen sequence records found in BOLD are derived from just a single marker (COI). Within BOLD, substantially fewer specimen records orginate from other mitochondrial markers like cytochrome \textit{b} (cyt\textit{b}) and the mitochondrial D-loop; even fewer come from nuclear gene regions such as ribosomal DNA (rDNA) and rhodopsin (rho). Thus, sequence reference databases should strive to incorporate genetic information from multiple genomic sources to better aid specimen identification to the species level, especially since the DNA barcode gap is nonexistent in most taxonomic groups outside animals \cite{kolter2020plant}. This is unfortunately easier said than done. Only quite recently have sequencing technologies such as Pacific Biosciences' SEQUEL platforms and genome skimming \cite{coissac2016barcodes} enabled the rapid and broad \\ characterization of biodiversity at multiple taxonomic levels. The widespread adoption of a multimarker approach for DNA barcoding has several limitations including the need for greater funding, as well as improved community standards. An ideal world is one where genomes for all taxa are available; however, currently only wealthy nations can afford to generate such massive amounts of data. This has led researchers within the community-at-large to forgo accomplishing goals ``the right way" at the expense of \\ upholding reproducibility. Global intiatives like Genome 10K \cite{G10KCOS2009genome, koepfli2015genome}, which specifically seeks to sequence the entire genomes of over 10000 vertebrate species, have provided biodiversity researchers with a glimpse into what will be possible once set data standards are adopted and strictly adhered to within the community-at-large \cite{koepfli2015genome}. The proposal for DNA standards such as that for obtaining high-quality DNA barcodes from \\ previously-collected specimens based on the reserved keyword ``BARCODE" \cite{hanner2009data} has been highly conducive in pushing DNA-based identification, albeit at a much smaller scale. Perhaps most importantly, Genome 10K points to a growing need for guidelines on the proper collection of specimens and the recovery of adequate amounts of specimen genetic information, the deep sequencing of specimen genomic DNA, as well as the timely deposition and curation of genome records to the International Nucleotide Sequence \\ Database Collaboration (INSDC) through publicly-accessible online molecular sequence databases such as the National Center for Biotechnology Information (NCBI)'s GenBank for easy retrieval, visualization and downstream analysis. 
 



\section{Discussion and Concluding Remarks}

In this piece, it was demonstrated that DNA barcoding currently lacks the \\ statistical rigor needed to properly interpret results of species barcode gap analyses through focusing on three key areas with respect to Metazoan taxa: (1) the need for larger specimen sample sizes reflective of standing genetic variation within species; (2) the misleading display of intraspecific and interspecific distances, and (3) the absence of formal statistical inference procedures in DNA barcoding. A past study of Pacific Canada's fish fauna by Steinke \textit{et al.}~ \cite{steinke2009dna} was employed to illustrate flaws in the presentation of the DNA barcode gap, as well as the need for larger specimen sample sizes to avoid biases in the reporting of within- and between-species genetic distances critical for reliably estimating the gap. First, the routine use of the novel R package {\tt HACSim} will allow researchers to better assess the efficacy of current taxon sampling schemes and develop more robust collection protocols that will permit greater statistical power in detecting a true species' barcode gap. Next, a more careful consideration of the depiction of the DNA barcode gap as a frequency histogram is warranted, as are alternative representations, including density estimation curves and the half-logarithm dotplot, due to interpretation issues surrounding default graphical parameters employed by many popular statistical analysis programs such as R and Excel. In addition, better ways to reconcile DNA barcoding with statistical inference include proposing the framing of the barcode gap as a one-tailed statistical \\ hypothesis test, and backing the use of the nonparametric bootstrap to compute standard errors and confidence intervals for maximum intraspecific distances, nearest-neighbour distances, as well as the barcode gap. Finally, new directions are offered for thinking critically about the robust estimation of the DNA barcode gap. While it is recognized that collecting specimens such as those of deepwater Pacific fishes is a costly and \\ time-consuming endeavour that  provide necessary genetic resolution deemed critical for large-scale biodiversity assessment, the introduction and proliferation of new quantitative methods to address knowledge gaps and uncertainties pertaining to the diversity of life on this planet will nonetheless make full use of the rapidly closing window of available sampling opportunity. Taken together, the methods outlined herein have the potential to open closed doors, giving biodiversity researchers and regulatory scientists an \\ unprecedented view of Earth's biodiversity over millions of years.



While focus was placed heavily on animal DNA barcoding and the importance of sound determination of the barcode gap, many of the principles that underlie taxon classification and demarcation algorithms mentioned or discussed in some detail herein are directly \\ transferrable to targeted species detection using DNA/eDNA as well as metabarcoding. Species-level discrimination is challenging (or unreliable) without extensive reference \\ libraries at hand. Thus, researchers are likely more concerned with pinpointing higher-level taxonomic matches. As a result, intraspecific variation and  barcode gap thresholds are largely ignored. In these instances, length variation in retrieved barcode sequences is widespread; thus, separate bioinformatic solutions are required like that of Barbera \textit{et al.}~\cite{barbera2019epang}. Despite this, efforts should be made to better integrate methods of DNA barcoding with (e)DNA metabarcoding, especially since many biodiversity researchers and regulatory scientists nowadays routinely employ elements of both disciplines in their work. 

    


With these considerations in mind, both biodiversity and regulatory scientists alike will be well-equipped to constructively analyze vast amounts of DNA barcode data with greater confidence and as a result feel more secure in making critical assessments as to the performance of DNA barcoding on the basis of the barcode gap. The widespread adoption of the methods discussed herein will be of great importance in moving forward with the building of large-scale DNA barcode reference libraries.
