\chapter{Incomplete estimates of genetic diversity within species: Implications for DNA barcoding}
\linespread{1.0}

\noindent Jarrett D. Phillips$^{1, 2}$, Daniel J. Gillis$^1$ and Robert H. Hanner$^{2, 3}$ \\ $^1$School of Computer Science \\ $^2$Centre for Biodiversity Genomics, Biodiversity Institute of Ontario \\ $^3$Department of Integrative Biology

\vspace{\fill}

\noindent An extended version of this chapter is published in \textit{Ecology and Evolution} \\ (Phillips \textit{et al.} \cite{phillips2019incomplete})


\section{Prologue}

At the time of publishing this work, little progress had been made in collating past studies on computational and statistical methods to estimate sample sizes for DNA \\ barcoding and other closely-related DNA-based identification techniques. This study \\ primarily touches on many important considerations of relevance to DNA barcoding of animal species in particular and aptly illustrates that full characterization of species genetic diversity is no easy feat. Most DNA barcoding studies published thus far have employed anywhere from 5-10 specimens per species to address research questions and objectives. However, a handful of studies point to the need for much larger specimen sample sizes, up to hundreds to thousands, required to adequately probe standing levels of intraspecific haplotype variation.



This work offers researchers a broad but focused overview of the challenges behind DNA barcoding. Importantly, in addition to striving to obtain much larger sample sizes than are currently in place, taxon sampling should be carried out across wide geographic regions whenever possible, subject only to factors such as budgetary and funding constraints. Broad geographic coverage of samples enables accurate and reliable downstream specimen identifications accomplished using well-populated barcode reference sequence libraries such as the Barcode of Life Data Systems (BOLD). To better aid in achieving this goal, the chapter closes with a short outline of traits future methods should seek to incorporate within their implementations.

   

Since its publication in February 2019, this paper has been cited 36 times according to Google Scholar.  


\begin{abstract}

DNA barcoding has greatly accelerated the pace of specimen identification to the \\ species level, as well as species delineation. Whereas the application of DNA barcoding to the matching of unknown specimens to known species is straightforward, its use for species delineation is more controversial, as species discovery hinges critically on present levels of haplotype diversity, as well as patterning of standing genetic variation that exists within and between species. Typical sample sizes for molecular biodiversity assessment using DNA barcodes range from 5-10 individuals per species. However, required levels that are necessary to fully gauge haplotype variation at the species level are presumed to be strongly taxon-specific. Importantly, little attention has been paid to determining appropriate \\ specimen sample sizes that are necessary to reveal the majority of intraspecific haplotype variation within any one species.


In this chapter, we present a brief outline of the current literature and methods on \\ intraspecific sample size estimation for the assessment of COI DNA barcode haplotype sampling completeness. The importance of adequate sample sizes for studies of molecular biodiversity is stressed, with application to a variety of metazoan taxa, through reviewing foundational statistical and population genetic models, with specific application to \\ ray-finned fishes (Chordata: Actinopterygii). Finally, promising avenues for further \\ research in this area are highlighted.

\end{abstract}


\section{Introduction}

One of the most fundamental problems underpinning much of modern molecular \\ biodiversity research is the issue of determining optimal levels of sampling effort that are required to adequately characterize biological sequence variation at the species level. Molecular genetic studies of biodiversity that utilize mitochondrial DNA (mtDNA) marker variation for the purpose of characterizing existing species genetic diversity are particularly sensitive to sample sizes. Four fundamental evolutionary forces act to alter the genetic composition of species populations: migration/gene flow, mutation, natural selection and random genetic drift. The effect of genetic drift on species populations is most evident when population sizes are small, as in the case of a recent bottleneck or founder event, resulting in the rapid loss of genetic diversity. Species differ in both their evolutionary histories and in their geographic distributions; therefore, the question of accurately \\ determining how many samples to include to observe a wide range of species genetic variation has been an ongoing area of interest and research. This is an important question deserving of more attention. Accurate determination of within-species (intraspecific) \\ sample sizes for mtDNA diversity estimation permits detailed analyses to be undertaken at the phylogenetic and phylogeographic levels to infer key biological processes such as isolation, dispersal and speciation \cite{avise1987intraspecific, dixon2006means, funk2003species}. Aside from addressing purely biological questions, the issue of determining optimal sampling strategies and sample sizes for genetic variation assessment at the species level also manifests at applied socioeconomic scales, particularly in the detection of food or natural health product fraud and in the monitoring of aquatic and terrestrial ecosystems \cite{hunter2015environmental}.



Within the field of biodiversity science, researchers have long recognized the \\ importance of sampling design to achieve a study's objectives. According to Lindblom \cite{lindblom2009sample}, well-developed sampling designs within the field of molecular biodiversity science should be formulated around three basic areas: research study questions, research study aims and taxonomic focus. In addition to these three areas,  Costa \textit{et al.} \cite{costa2015conservation} point to further considerations: planning the number and geographical distribution of specimens to be sampled, the category and number of genetic loci to be examined, and the spatial distribution and number of individuals to be sampled within each species' population. While there is a lack of clear sampling guidelines currently in place for optimal \\ spatio-temporal assessment of species populations, Pante \textit{et al.} \cite{pante2015species} argue that such schemes should be guided by adequate coverage of both the putative geographic/ecologic range of the species under study, as well as potentially closely related species over their entire range. Given that much of species spatio-temporal metadata is not reported alongside genetic data, such assessments become problematic unless community standards and \\ practices are improved \cite{hanner2009data, naaum2015standards, strohm2016mitogenome}. Where this becomes particularly important is in the development and design of species-specific real-time Polymerase Chain Reaction (qPCR) primers and probes, for integration within environmental DNA (eDNA) assays for instance. This is especially the case if such tools are to be continuously implemented within 
\\ regulatory or forensic settings such as the Canadian Food Inspection Agency (CFIA) \cite{Shehata2018dna} and the United States Food and Drug Administration (USFDA), as the success of such methods depends greatly on the extent of geographic coverage of species genetic diversity.

 

The overall goal of sampling is to make inferences concerning a population of interest based only on information contained within finite samples drawn from the larger \\ population. This is done though estimating population parameters such as the population mean ($\mu$) using the sample mean ($\bar{x}$). One example, relevant to molecular population genetics, is the calculation of average pairwise distances based on Nei's estimator of \\ nucleotide diversity ($\pi$) \cite{nei1979mathematical}. Under the Frequentist statistical paradigm, the minimum sample size that is required to estimate a population mean, from a Normal distribution, is given by \cite{adcock1997sample}

\begin{equation}
n \geq \left(\frac{z_{\alpha/2}\sigma}{d}\right)^2
\end{equation}



\noindent where $z_{\alpha/2}$ is the appropriate critical value to estimate $\mu$ with a level of significance of $1 - \alpha$, $\sigma^2$ is the population variance and $d$ is the desired margin of error. From the above equation, the required minimum sample size is controlled by the experimenter through the margin of error. A smaller margin of error results in a larger value of $n$. Similarly, predicting $n$ with a higher level of accuracy can be achieved through narrowing $d$. Sample sizes that are computed from the above equation serve as a baseline requirement prior to conducting any quantitative study of interest. Depending on the sampling scheme, for instance stratified sampling, other formulas exist for the appropriate calculation of necessary sample sizes.



In determining the most appropriate sample size required for a particular study, a crude rule of thumb that is often used in statistics and other scientific disciplines pertains to the use of a sample size of at least $n = 30$ when making comparisons among study groups or when deciding to use probabilities derived from the Standard Normal distribution \cite{cohen1990things}. Unfortunately, adequate sample sizes, while widely viewed as being central to a given biodiversity research study, are often neglected in practice \cite{lenth2001some}. In such cases, this may be due to, for example, costs associated with or resources required for adequate specimen collection \cite{cameron2006will, hortal2005ed, muirhead2008identifying}.



Statistical power analysis can be employed to help shed light on sample sizes required to detect a given effect prior to carrying out a scientific study. Power, which is defined as the complement of the Type II error rate ($\beta$), depends on four factors: effect size (ES), significance level/Type I error rate ($\alpha$), sample size ($n$) and population standard deviation ($\sigma$) through the proportionality \cite{di2003much}

\begin{equation}
(1 - \beta) \propto \frac{\textrm{ES} \times \alpha \times \sqrt{n}}{\sigma}.
\end{equation}

\noindent Effect size is the difference between an observed quantity and one hypothesized under a null distribution. Larger deviations lead to greater power to detect real effects. It is easily seen from the above proportionality that larger values of effect size, significance level and sample size all generate higher levels of statistical power; whereas, increasing population standard deviation results in loss of power. Together with the sample size equation discussed previously (Equation 1), many factors are at play in determining the most appropriate sample size needed for a given study.



Any sampling scheme that is carried out will be subject to systematic error. Sampling (ascertainment) bias is an important factor to consider in this regard because it can lead to under- or overestimation of population parameters. Ascertainment bias describes the tendency of certain individuals to be less likely sampled than others \cite{parr2012evolutionary} and is common in molecular biodiversity studies (\textit{e.g.}, \cite{hanner2011fish, muirhead2008identifying, mutanen2016species, wilkinson2017replacing}). This can occur, for example, when sampling is restricted to certain geographic regions \cite{muirhead2008identifying} or to particular species (\textit{e.g.}, those known to be of conservation importance) \cite{hanner2011fish}. Sampling bias can be minimized through increasing the geographic breadth of a study, in addition to targeting representative taxa with large specimen sample sizes.



The present review briefly examines current approaches for species genetic variation assessment as it relates to the estimation of intraspecific sample sizes for DNA barcoding. Specifically, the focus will be on COI DNA barcode haplotype sampling completeness. Few studies have focused on DNA barcode sample size prediction for wide-ranging taxa in this regard. Here, methods of haplotype variation assessment are first covered. This is then followed by an examination of existing studies, with particular consideration of important findings to date within the literature. Finally, promising new avenues for further research are explored.



\section{Current Methods}

\subsection{Methods to Assess Haplotype Variation}

\subsubsection{Haplotype Diversity}

Genetic diversity is manifested within species in several ways. One way is through haplotype variation. While there are many different definitions of what constitutes a \\ haplotype, in the broadest sense, a haplotype is a unique DNA sequence that differs from others at one or more basepair positions within and between species. Nei's \cite{nei1987molecular} haplotype diversity ($h$), which is a widely-used approach to measuring genetic variation within \\ species populations, is given by the equation 

\begin{equation}
h = \frac{n}{n-1}\left(1 - \sum_i{p_i^2}\right).
\end{equation}



\noindent where $p_i$ is the (normalized) frequency of the $i^{th}$ haplotype in the sample. Two \\ interpretations of \textit{h} are that it expresses the probability of observing a previously unseen haplotype upon sampling a new individual \cite{wares2015can} or that it represents the probability that two haplotypes, selected at random from a sample of $n$ DNA sequences, are distinct \cite{goodall2012comparison}. Haplotype diversity can also be quantified using the absolute number of haplotypes ($H$). Both $h$ and $H$ are greatly affected by levels of sampling intensity within species. In particular, undersampling can cause these measures to become under- or overestimated \cite{goodall2012comparison}. Several other approaches are in wide use to aid researchers in assessing levels of standing genetic variation existing within species populations. Two of these are haplotype networks and haplotype accumulation curves. 



\subsubsection{Haplotype Networks} 

A widely-used approach to assessing levels of genetic variation within and between species is through the construction of haplotype networks \cite{templeton1992cladistic}. Haplotype networks \\ accurately represent differences existing among sampled haplotypes through grouping \\ identical DNA sequences within the same vertex. The size of a given vertex is proportional to the number of DNA sequences it contains.  Divergent haplotypes are connected via edges that display the number of mutational differences separating adjacent vertices.

 

Haplotype networks are appealing because they can be used to infer potential cryptic diversity within a taxon or interspecific hybridization between allopatric \\ (\textit{i.e.}, reproductively-isolated) species, but interpretation can sometimes become difficult when multiple species cluster together into one or multiple nodes or subnetworks \cite{hanner2011dna, hart2007things, wong2009identifying} or when ambiguous/missing nucleotide data are present within DNA sequences (\textit{e.g.}, Ns or gaps (--)) \cite{joly2007haplotype}. While haplotype networks cannot give a direct indication of the level of sampling completeness for a given species, the presence of numerous rare haplotypes suggests gross undersampling of intraspecific genetic variation (or alternatively PCR/sequencing error).




\subsubsection{Haplotype Accumulation Curves}

Assessing the completeness of intraspecific haplotype sampling can be carried out through generating haplotype accumulation curves. Such curves are analogous to \\ rarefaction curves used in studies of species richness \cite{gotelli2001quantifying} and depict the degree of \\ asymptotic behaviour as a function of both the number of specimens sampled and the cumulative mean number of haplotypes accumulated. Initially, accumulation curves will increase very rapidly since many new haplotypes will be captured for a given species with minimal sampling effort, but haplotype recovery slows drastically as sampling depth is increased because many haplotypes that are found will have already been observed previously. Thus, species curves showing rapid saturation strongly suggest that the majority of haplotype diversity has been uncovered; whereas, those curves displaying little to no evidence of reaching an asymptote indicate that further sampling is required \cite{zhang2010estimating}. \\ Deciding whether a species should be further sampled can be deduced from the magnitude of the slopes calculated using a fixed number of points occurring on the end of the curve (e.g., ten in the case of \cite{phillips2015exploration, young2012revealing}). Slopes near or below a predefined threshold, for example, 0.01 (\textit{i.e.}, equivalent to observing one new haplotype for every 100 DNA \\ sequences), suggest that additional sampling is unlikely to reveal any  new haplotypes; whereas, those species curves with slopes above 0.1 (\textit{i.e.}, observing one new haplotype for every 10 DNA sequences) strongly indicate that further sampling is necessary \cite{hortal2005ed}. 



One obvious problem that arises in the use of haplotype accumulation curves to gauge species genetic diversity and levels of sampling effort, however, is the fact that the \\ functional form of such curves is not known and can differ widely across taxa \cite{phillips2015exploration}. \\ Furthermore, deciding on appropriate curve slope thresholds necessary for adequate \\ sampling coverage are largely arbitrary \cite{hortal2005ed}. While various parametric model \\ curve-fitting approaches, such as the power, negative exponential and Michaelis-Menten functions, have been heavily employed and debated in the literature to model species-area relationships \cite{dengler2009function, tjorve2003shapes} or species richness, no single approach yet exists that can be readily applied to determine sample sizes that are likely required for intraspecific genetic variation \\ assessment.



A second, lesser-investigated issue, relates to the fact that haplotype accumulation curves are not spatially-explicit. Thus, it becomes difficult to account for correlations that may exist at the subpopulation or higher taxonomic levels. This has been noted in past studies of species richness employing species accumulation and rarefaction curves (\textit{e.g.}, \cite{bevilacqua2017approach, chiarucci2009spatially, terlizzi2014species}).



\subsection{Sampling Models for Genetic Diversity Prediction}

In addition to qualitative approaches to assessing standing genetic variation within \\ species, a number of quantitative models to estimate required sample sizes for overall genetic diversity assessment have been proposed. These include Frequentist, Bayesian and coalescent models.



Holt \textit{et al.} \cite{holt2007experimental} reviewed several Frequentist and Bayesian statistical methods of sample size determination for intraspecific haplotype diversity assessment that are most \\ informative over large geographic ranges. The authors note that a lower bound on the probability of sampling a dominant haplotype in a sample of size $n$ with significance level $\alpha$ is given by the inequality

\begin{equation}
p \geq \sqrt[n]{\alpha}.
\end{equation}



Grewe \textit{et al.} \cite{grewe1993mitochondrial} employed a binomial sampling model to determine the minimum sample size required to assess mtDNA variation in Lake Ontario lake trout (\textit{Salvelinus namaycush}) stocks according to the equation

\begin{equation}
n=\frac{\textrm{ln}(1 - \beta)}{\textrm{ln}(1 - p)}
\end{equation}



\noindent where \textit{p} is the frequency of a given haplotype and $\beta$ is the desired confidence level. The authors found that $n$ = 60 individuals are likely needed to be randomly sampled to observe a single haplotype having a frequency of at least $p$ = 5\% with $\beta$ = 95\% confidence. It is worth noting that this figure increases to \textit{c}. 460 individuals for a haplotype occurring at frequency of 1\% with 99\% confidence \cite{grewe1993mitochondrial}. This marked increase in sample size is not surprising given that one would need to sample many more individuals to be certain that the majority of rare haplotypes have been uncovered. It is important to note, however, that Grewe \textit{et al.} \cite{grewe1993mitochondrial} sampled individuals from six different and highly divergent trout strains (\textit{i.e.}. local populations), each displaying high degrees of population substructure. Population subdivision likely will have an effect on the estimation of required sample sizes needed to gauge levels of standing genetic variation at the species level. 



Similar magnitudes of sample sizes were found by Austerlitz \textit{et al.} \cite{austerlitz2009dna}, who employed coalescent theory \cite{kingman1982coalescent}, to determine the probability of adequately sampling all genetic variation of a species with sample size $n$. Coalescent theory attempts to trace the lineage of an ancestral allele (termed the Most Recent Common Ancestor, MRCA) backwards in time within a gene genealogy. Under a geometric distribution, this probability is given by the equation \cite{austerlitz2009dna}

\begin{equation}
p = \frac{n - 1}{n + 1}.
\end{equation}



\noindent From the above equation, only $n$ = 39 individuals are required to be sampled at random to observe $p$ = 95\% of all genetic diversity for a species. It should be noted however that even with increasing sample sizes, one's confidence in having sampled all of a speciesâ€™ genetic diversity approaches closely, but never actually reaches, 100\% \cite{austerlitz2009dna}. This is illustrated by the finding that the required sample size increases to $n$ = 1999 individuals necessary to observe $p$ = 99.9\% of the total genetic diversity that exists for a given species using Equation (2.6). This can be explained by the fact that individual haplotypes for a given species become much more difficult to recover as the intensity of specimen sampling is increased because intraspecific genetic variation is expected to increase as a result. The coalescent, as a large-scale sampling model, has found wide application in DNA-based approaches to species identification and delimitation, most notably DNA barcoding \cite{hubert2015dna}.



\subsection{DNA Barcoding}

Since its inception in 2003, DNA barcoding \cite{hebert2003biological} has risen to become the largest \\ taxonomically-driven biodiversity initiative to date aimed at identifying and cataloging all assemblages of multicellular life on the planet. DNA barcoding is a genomic technique that relies on DNA sequence variation within short, standardized gene regions to rapidly identify specimens to the level of species and to discover new species. The ideal DNA barcode is one that is found in all organisms, readily distinguishes between taxa, and is easily amplified, sequenced and aligned. In animals, the agreed-upon marker of choice for taxon assignment is a \textit{c.} 650 basepair (bp) fragment from the 5' end of the \\ mitochondrially-encoded cytochrome \textit{c} oxidase subunit I (COI) gene. Mitochondrial loci like COI are particularly suitable as genetic markers for DNA barcoding because they are fast evolving, highly conserved across taxa, present in high copy number, haploid, maternally inherited, lack introns, display few insertion-deletion (indel) mutations, and experience little to no gene recombination \cite{hebert2003biological, hebert2003barcoding}.



The primary goal of DNA barcoding has been to develop a publicly accessible species reference sequence library to aid in the identification of unknown specimens and accelerate the discovery of potentially undescribed taxa. Obtaining adequate sample sizes for building accurate and reliable specimen reference libraries has culminated in the development of the Barcode of Life Data Systems (BOLD; http://www.boldsystems.org) \cite{ratnasingham2007bold} as the largest collection of user-curated species sequence data specifically for DNA barcoding currently available on the World Wide Web. At present (as of May 1, 2018), BOLD holds over six million DNA barcode records from over 250,000 named species. Certain taxa are well represented in BOLD with upwards of hundreds of barcode sequences for some species. Despite this, barcode reference libraries within BOLD remain largely incomplete, even for the most well-sampled taxa such as fishes and insects. As such, comprehensive coverage of species genetic diversity is still decades away \cite{wilkinson2017replacing}. Wilkinson \textit{et al.} \cite{wilkinson2017replacing} points to strong ascertainment bias as the most likely explanation for this. In the early days of BOLD, DNA barcode sequence acquisition was high, due to the fact that over 75\% of taxon records were mined from already well-established sequence databases such as GenBank \cite{wilkinson2017replacing}.



\subsection{The Importance of Sampling to DNA Barcoding}

DNA barcoding works in practice because interspecific (between-species) variation is usually much greater than intraspecific (within-species) divergence \cite{meyer2005dna, stoeckle2014dna}. While this observed `barcoding gap' \cite{meyer2005dna} is a necessary criterion for successful taxonomic resolution using distance-based methods, it may not be a sufficient one for other molecular approaches (\textit{e.g.}, those employing tree- or character-based techniques). Cases are well documented where considerable overlap/separation between (maximum) intraspecific variation and  \\ (minimum) interspecific divergence exists \cite{hebert2004identification, hubert2015dna}. Undersampling can greatly \\ exaggerate the existence of the barcode gap. The inclusion of small sample sizes over large geographic ranges has the effect of obscuring existing mitochondrial sequence diversity at the species level since the finding of divergent haplotypes may be the result of poorly sampled panmictic (\textit{i.e}, randomly-mating) intraspecific variation \cite{clare2011neotropical}. Compared to \\ regional scales, with increasing sampling effort across wider spatial scales, intraspecific variation is expected to increase whereas interspecific divergence will decrease in effect since more closely-related species will tend to be found due to allopatric speciation being a dominant mode of diversification \cite{bergsten2012effect, pentinsaari2014barcoding}. 



How much variation is actually needed to separate species is not known with certainty because intraspecific sampling has generally been limited to narrow geographic locales. Hebert \textit{et al.} \cite{hebert2003biological} proposed that barcode sequences exhibiting at least 2\% nucleotide \\ divergence should be designated as being from distinct species. Intraspecific distances larger than 2\% suggest the presence of cryptic species, whereas those smaller than 2\% is evidence for evolutionarily young species with a recent origin (\textit{i.e.}, retention of ancestral polymorphisms due to incomplete lineage sorting), hybridization/introgression or \\ inadequate taxonomy (\textit{e.g.}, cryptic species or species synonymy) \cite{hubert2015dna}. In BOLD, query sequences are matched to reference barcodes based on a genetic distance heuristic of 1\% \cite{ratnasingham2007bold}. The use of such threshold estimates for species separation is arbitrary and is often applied to a wide variety of taxa, regardless of species life histories. A later estimate of ten times the mean intraspecific distance (the so-called `$10\times$ rule') was given by Hebert \textit{et al.} \cite{hebert2004identification}. Unlike the previously suggested estimate of 2\% sequence divergence, the $10\times$ rule makes use of all available taxon sequences within a dataset to calculate an appropriate limit for species separation. Despite this, the $10\times$ rule has been met with criticism: Collins and Cruickshank \cite{collins2013seven} suggest consideration of the maximum intraspecific distance and the minimum interspecific divergence (\textit{i.e.}, nearest neighbour distance) for each species under investigation. The use of lower generic thresholds for species discovery may falsely inflate existing genetic diversity, whereas the adoption of higher cutoffs would likely be too conservative for reliable detection of cryptic species \cite{april2011genetic}. It is well understood however that the most appropriate cutoff necessary to accurately diagnose species on the basis of sequence variation is strongly taxon-dependent \cite{hebert2003barcoding, hickerson2006dna, meyer2005dna} and will become more precise with increased sampling effort.



DNA barcoding has its roots in the historic disciplines of Darwinian evolutionary \\ theory, population genetics and phylogenetics: the coalescent is a modern interpretation that reconciles these domains \cite{rosenberg2002genealogical}. While genetic distance-based approaches to species \\ delineation are commonplace within barcoding studies because they scale well to large taxon datasets, early-proposed arbitrary separation methods like the 2\% or 10$\times$ rule \\ completely ignore evolutionary relationships that exist among closely-related species. \\ Objective tools for the delineation of species are well known and generally fall into three overlapping categories: phylogenetic, coalescent, and phylogenetic-coalescent \cite{hubert2015dna}. The well-known neighbour-joining clustering method was advocated for in the early barcoding literature as a means of confirming the presence of reciprocal monophyly across sampled taxa. More recently, novel bioinformatic algorithms, most notably distance-based \\ approaches such as Automatic Barcode Gap Discovery (ABGD; \cite{puillandre2011abgd}) and tree-based \\ methods including variants of the Generalized Mixed Yule Coalescent (GMYC; \cite{monaghan2009accelerated, pons2006sequence}) have been put forth to facilitate species separation, an otherwise daunting task for even the most highly-skilled and knowledgeable taxonomist. ABGD is a nonparametric technique of partitioning species on the basis of the barcode gap using DNA sequences. On the other hand, GMYC is a likelihood-based method that relies on the premise that bifurcation (\textit{i.e}, fully-resolved branching) within ultrametric species trees is indicative of \\ speciation/diversification events, and therefore suggests the presence of undescribed taxa.  A key factor in the success of such methods is sample size, and few groups have been so extensively inventoried \cite{hubert2015dna}. For example, GMYC is especially prone to the under- or overestimation of putative species, which can be magnified due to differences in effective population sizes as well as historical versus contemporaneous patterns of migration/gene flow among subpopulations     \cite{lohse2009can, papadopoulou2009sampling}. Thus, sufficient sampling is paramount. Often, \\ researchers would like to know whether all unique haplotypes within a lineage or deme have been adequately sampled; unfortunately, this is complicated by the fact that the majority of species are both geographically-widespread and rare. As a result, given that ascertainment and operational biases are inevitable \cite{mutanen2016species}, an extensive sampling of all local populations that comprise a given species is unrealistic, even under the best situations (\textit{e.g.}, strong research budget, easy access to sampling locations). Thus, whenever possible, a more comprehensive sampling of study sites is required in order avoid false positives/negatives and to reveal divergent haplotypes that may have been missed with spatially-narrower sampling routines \cite{monaghan2009accelerated}. Incorporation of coalescent and population genetics theory can aid in informing researchers on broad macro-level processes that may be at play in shaping trends seen within haplotype accumulation curves on the basis of extant patterns of \\ intraspecific genetic diversity. 


 
The Barcode Index Number framework for animals, first introduced by Ratnasingham and Hebert \cite{ratnasingham2013dna}, represents a novel approach to addressing the issue of sample sizes necessary for barcoding initiatives. The BIN system partitions COI barcodes into distinct Operational Taxonomic Units (OTUs) on the basis of the REfined Single Linkage (RESL) clustering algorithm and Markov clustering \cite{ratnasingham2013dna}. BINs comprise high-quality sequences linked to BARCODE compliant records. The BARCODE standard currently in place stipulates that only barcode sequences with read lengths of at least 500 bp and containing less than 1\% ambiguous nucleotides are designated unique BIN clusters \cite{hanner2009data}. While BINs generally show high concordance with actual biological species, they can be further 
\\ employed to gauge instances of suspected cryptic species diversity, especially in the cases where intraspecific distances are not clearcut.  Species that fall into two separate BINs (termed a SPLIT) is evidence that they are being overlumped. Further, the occurrence of rare BINs (\textit{i.e.} those represented by a single specimen) may be the result of limited sampling \cite{hausmann2013genetic, huemer2014testing}. Stand-alone BINs may also reflect sequencing errors in the form of very low-frequency (VLF) variants or cryptic pseudogenes \cite{stoeckle2012frequency, stoeckle2014dna}. Increased sampling coverage can be beneficial in such instances, as true biological variation is less likely to be misidentified as artificial biological variation and unintentionally flagged as potential VLFs.






\section{Key Findings}

\subsection{DNA Barcoding and Sample Size: Past Studies}

The ability of DNA barcodes to uncover levels of standing genetic variation within species is strongly influenced by the scale of specimen sampling, which has been \\ recognized as a major barrier to the success of DNA barcoding since its early days  \cite{hebert2004identification, meyer2005dna, ward2005dna}. In spite of this, global barcoding efforts have only been partially successful in capturing the full extent of COI barcode variation in animals due to the majority of studies forgoing deep taxon sampling in favour of maximizing the number of different taxa sampled \cite{matz2005likelihood, zhang2010estimating}. Sample sizes of a few individuals per species (typically in the range of 5-10, but one or two specimens is not uncommon since these are often the only representatives available). This could be either due to unclear species boundaries  or limited geographic sampling of intraspecific variation, Such sample sizes and explanations are widespread in barcoding studies \cite{hajibabaei2007dna, matz2005likelihood, zhang2010estimating}. Recommended sample sizes currently in place are by no means sufficient since species abundance is often skewed \\ geographically/ecologically. For example, five specimens per species per FAO (Food and Agriculture Organization) region was initially suggested by the Fish Barcode of Life \\ (FISHBOL; \cite{ward2009campaign}) initiative, but the sampling of up to 25 individuals or more may be necessary for some species exhibiting widespread distribution patterns \cite{becker2011five, steinke2011fish}. Similarly, in assessing haplotype and nucleotide COI variation across wide-ranging animal taxa, Goodall-Copestake \textit{et al.} \cite{goodall2012comparison} note that a sample size of five individuals per species population was adequate to differentiate between extremes of $h$, but as many as 25 \\ specimens would need to be collected to achieve maximum accuracy. Jin \textit{et al.} \cite{jin2012simple}, and Matz and Nielsen \cite{matz2005likelihood} both point to a sample size of 12 specimens, whereas Ross \textit{et al.} \cite{ross2008testing} suggest that sampling five or more reference barcodes is sufficient for accurate species identification. Bias toward low sample sizes observed for most species may be the result of many factors (see Bucklin \textit{et al.} \cite{bucklin2011dna} for a concise summary in marine metazoa), including the presence of cryptic diversity, amplification of non-functional gene copies (\textit{i.e.}, pseudogenes/nuclear-mitochondrial inserts (NUMTs)), contamination by foreign DNA \\ from other species (\textit{e.g.}, bacterial symbionts such as \textit{Wolbachia}), insertion-deletion (indel) mutations, or errors arising from PCR/sequencing runs \cite{goodall2012comparison}. \\ Molecular diagnosis of specimens to the species level using DNA barcoding is not \\ definitive; numerous technical sources of error exist that can hamper the ability of reliable taxon assignment, in particular, misidentifications, sequencing errors and lack of taxonomic metadata (\textit{e.g.}, inclusion of GPS coordinates, record linkage to a voucher specimen). While such factors are likely to occur infrequently for interspecific barcodes, this is not the case for intraspecific datasets. Taken together, biases in sample sizes  will likely be considerable. In certain cases, the occurrence of biological phenomena can lead to problems encountered later on in the lab, specifically during the sequence amplification stage using PCR. A well-known example of this is the symbiotic association of the bacterium \textit{Wolbachia} with insects. Integration of \textit{Wolbachia} within host genomes of various Hymenoptera, Diptera and Lepidoptera can cause fluctuations in intraspecific distances \cite{smith2012wolbachia} and thus, observed haplotype diversity between infected and uninfected hosts \cite{chen2017effects}. Misamplification of host sequences for bacterial symbionts is widely encountered, as is the amplification of \\ pseudogenes/NUMTs. Technical sources of error such as expert taxonomic \\ misidentifications, sequence contamination, as well as errors arising from the \\ amplification/sequencing process can be controlled, and can be minimized to a degree. Two critical steps in avoiding such issues are: (1) the construction of an NJ tree to pinpoint potentially misidentified specimens and/or sequence contaminants (as opposed to solely being used in the establishment of reciprocal monophyly, as argued by Collins and \\ Cruickshank \cite{collins2013seven}) and (2) the careful inspection of BOLD specimen trace files to resolve noisy sequence regions that inflate estimates of standing genetic variation through the introduction of functional (heteroplasmic) sequence variation (as in \textit{e.g.}, Hebert \textit{et al.}, \cite{hebert2004ten}) and/or nonexistent low-frequency species haplotypes occurring in high abundance \cite{stoeckle2012frequency}. The effect of these on generated haplotype accumulation curves is delayed saturation to an asymptote due to larger required sample sizes. Combined with initially large numbers of specimens within intraspecific data sets (\textit{e.g.}, $N > 100$), this effect can be quite substantial. As BOLD is ever-evolving, in part due to the sheer volume of DNA barcode sequences being added on a daily basis, it is crucial that suspected errors within taxon records be dealt with in a timely manner (\textit{e.g,}, through community users flagging problematic records for closer examination by submitters), so that sequence integrity is not compromised. While the issue of determining adequate sample sizes for molecular species diagnosis has largely been aimed at animal taxa, Liu \textit{et al.} \cite{liu2012sampling} explored optimal sample sizes needed for plant DNA barcoding. It was found that relatively small sample sizes were adequate to recover sequence variation in slowly evolving genes (2 or 3 sequences per species population for matK); whereas, higher numbers are necessary for rapidly evolving markers (minimum of 10, 8 and 6 individuals per population for trnH-psbA, trnL-trnF and ITS respectively) \cite{liu2012sampling}. Further, the authors found that a sample size of 8-10 individuals per species across the entire geographic range appears sufficient for \textit{Taxus} barcoding. Unfortunately, such small sample sizes, likely the result of low information content due to the high presence of sequence artifacts (\textit{e.g.}, indels within mitochondrial/plastid markers), often lack discriminatory \\ power that is needed for accurate identification of specimens on the basis of genetic \\ polymorphism with DNA barcodes.

 

To date, few studies explicitly exploring simulated sample sizes for DNA barcoding in wide ranging animal taxa have been conducted. One of the first studies to examine the issue of sample sizes for DNA barcoding via haplotype accumulation curves was conducted by Zhang \textit{et al.} \cite{zhang2010estimating} using a modified form of the Michaelis-Menten equation. Using this method, the authors found that the random sampling of 250-1188 individuals from the Costa Rican skipper butterfly (\textit{Astraptes fulgerator}) cryptic species complex are likely needed to detect 95\% of all genetic diversity for this species based on an initial sample size of 407 individuals. Conversely, the same authors found that 156-1985 specimens were needed to retrieve 95\% of COI variation using simulated island \cite{wright1951genetical} and stepping-stone \cite{kimura1964stepping} coalescent models across three distinct subpopulations and under varying effective population sizes. In addition, a sample size outlier of only 47 individuals was found for one subpopulation of \textit{A. fulgerator} butterflies.  The authors note that this may be due to the low level of genetic variation observed in this population: only two haplotypes were observed across 14 sampled individuals. In contrast, a later study on European diving beetles undertaken by Bergsten \textit{et al.} \cite{bergsten2012effect} found that based on 419 sampled \textit{Agabus bipustulatus} specimens, a sample size of 250 specimens was required to be randomly sampled across its range to achieve 95\% haplotype recovery. On the other hand, 70 \\ individuals of the same species was necessary to be sampled to recover 95\% of COI variation when geographic dispersion between a new sample and the closest previous sample was maximized using resampling simulation. 



Not all studies find evidence for greatly broadening the scope of comprehensive \\ specimen sampling. Luo \textit{et al.} \cite{luo2015simulation} demonstrate the utility of the Central Limit Theorem (CLT), employing a simple resampling scheme along with the modified Michaelis-Menten \\ saturation model. The CLT states that the distribution of the sample mean tends toward the (Standard) Normal distribution as the sample size increases. It was found that a minimum sample size of only 20 individuals is needed to provide a reliable estimate of genetic polymorphism at the species level on the basis of observed haplotype numbers. The authors note however that sample sizes should be as large as possible, even though new haplotypes will tend to be observed with lower frequency. Compared to present sample size range of 5-10 specimens per species, a slightly larger minimum sample size range of 11-15 individuals per species was recommended by Yao \textit{et al.} \cite{yao2017evaluating} for widely-distributed coastal and inland aquatic salt-tolerant plant species of the families Poaceae and Chenopodiaceae across seven different genera, based on results obtained through resampling procedures and \\ nonparametric Mann-Whitney $U$ tests.



Though not devoted to estimating sample sizes for mitochondrial genes such as COI, using resampling simulation, Hale \textit{et al.} \cite{hale2012sampling} found that a sample size of 25-30 individuals was sufficient to accurately estimate microsatellite allele frequencies in hypothetical \\ populations of hairy wood ants (\textit{Formica lugubris}), kakis (\textit{Himantopus novaezelandia}), black-browed albatrosses (\textit{Thalassarche melanophris}) and red squirrels (\textit{Sciurus vulgaris}). The sampling of 25-30 individuals per species for the assessment of genetic diversity via microsatellite loci was also recommended by Pruett and Winker \cite{pruett2008effects} in an earlier study of song sparrows (\textit{Melospiza melodia}). A more recent simulation study examining minimum sample sizes for accurate estimation of genetic diversity from a large number of single nucleotide polymorphism (SNP) markers in the terrestrial Amazonian plant \textit{Amphirrhox longifolia} found that sample sizes beyond eight are sufficient for genetic diversity \\ assessment and as few as two individuals are needed to obtain good estimates of population differentiation \cite{nazareno2017minimum}. These studies clearly point to the need for large sample sizes in multilocus population genetic studies for the overall assessment of genetic diversity at the species level.



These examples serve to illustrate the fact that, as is the case for species divergence thresholds, there is no one universal sample size that can accurately recover the majority of intraspecific genetic variation across taxa and it appears likely that varying levels of additional sampling will be required within taxa and across geographic ranges \cite{lou2012effect}. What seems to be clear is the fact that many previous assessments of sample sizes necessary for DNA barcoding studies have underestimated levels of sampling depth that are actually needed to recover much of the genetic variation that exists at the species level. Such a trend seems most attributable to restricted geographic sampling and unclear species boundaries, limited funding for adequate specimen retrieval, as well as human-mediated mechanisms such as errors accrued during the amplification/sequencing process. 

\subsection{Application to Ray-finned Fishes}

Phillips \textit{et al.} \cite{phillips2015exploration} investigated levels of existing COI haplotype variation in 18 species of ray-finned fishes (Chordata: Actinopterygii) represented by a minimum of 60 individuals in accordance with Grewe \textit{et al.} \cite{grewe1993mitochondrial}. Results showed that 147-5379 specimens likely must be randomly sampled to uncover all predicted haplotype diversity in the selected species (between 3-528 total haplotypes) \cite{phillips2015exploration}. Sample size estimates obtained by Phillips \textit{et al.} \cite{phillips2015exploration} are comparable in magnitude to those of Zhang \textit{et al.} \cite{zhang2010estimating}, but not in the case of Luo \textit{et al.} \cite{luo2015simulation}, which are closer to practical sample sizes for DNA barcoding. Further, haplotype accumulation curves displayed evidence of reaching an asymptote for only 3/18 examined species: Chinook salmon (\textit{Oncorhynchus tshawytscha}), Rockfish (\textit{Sebastes} sp.) and Siamese fighting fish (\textit{Betta splendens}) based on significance testing of curve slopes with a one-sided \textit{t}-test using the last 10 points on the end of accumulation curves \cite{phillips2015exploration}. Of note is the haplotype accumulation curve for Chinook salmon, which appeared to show premature saturation despite only 12 out of an estimated total of 78 haplotypes being found for the species. At the time of publication of Phillips \textit{et al.}'s \cite{phillips2015exploration} study, \textit{Sebastes} sp. was linked to a single BIN. The BIN system is inherently dynamic: as more sequences are added within BOLD, specimens assigned to a single BIN may be allocated to multiple BINs or multiple existing BINs may be coalesced into a single BIN. This is especially the case as species boundaries become clearer or taxonomic revisions are made. As an example, the genus \textit{Sebastes} is a highly speciose group, thought to have undergone an adaptive radiation as recently as 8-9 million years ago \cite{steinke2009dna}. This fact could explain the low haplotype diversity observed for this species (two haplotypes across 98 individuals).

Determining the precise point corresponding to haplotype accumulation curves \\ reaching an asymptote (\textit{i.e.}, having a slope near zero) is difficult. One way this can be accomplished is through employing numerical techniques, specifically iteration. Such methods work by repeatedly recycling computed values into an algorithm; that is, current values are used as starting values to the next iteration until convergence to a solution is achieved. One way this can be realized is through iterating Equation (7) along with the equations for the ``measures of sampling closeness'' proposed by Phillips \textit{et al.} \cite{phillips2015exploration}. This seems to be the most logical way forward in better ascertaining at what level specimen sampling is deemed sufficient and thus, when further collection of specimens should be ceased.

\section{Future Prospects}

This chapter explores the issue of sampling in DNA barcoding from the perspective of computational and statistical methodologies. Key sample size studies in the barcoding literature were examined in detail. A lack of consensus exists in the most appropriate number of specimens that must be targeted to uncover the majority of haplotype diversity that exists at the species level for a variety of taxa. This question is similar to the problem of calculating species divergence thresholds for taxon delimitation and is strongly dependent on species abundances, life histories and geographic coverage. To date, few studies \\ exploring sample sizes for DNA barcoding have been conducted. Existing studies (\cite{phillips2015exploration, zhang2010estimating}) appear to point to the comprehensive sampling of hundreds to thousands of specimens to capture a wide range of standing genetic variation for a given species based on \\ asymptotic behaviour of haplotype accumulation curves.

Future research should aim to answer the question: is there an optimal threshold for specimen sampling above which no new DNA barcode haplotype variation is likely to be observed for a species? While it should be possible to find this limit for already well-sampled taxa based on trends seen in haplotype accumulation curves, the use of haplotype accumulation curves to estimate sample sizes that are required for full \\ assessment of COI DNA barcode haplotype sampling completeness has only been tested in one previous study (Zhang \textit{et al.} \cite{zhang2010estimating}). Phillips \textit{et al.} \cite{phillips2015exploration} expanded on previous studies through proposing a simple and easily implemented method to estimate specimen sample sizes for a number of ray-finned fish species, which are among the most densely sampled to date within BOLD. Sample size optimization for the identification of animal species across wide-ranging geographic scales is key since intraspecific variation within DNA barcodes is not easy to measure, and obtaining large numbers of barcodes that reflect a wide range of intraspecific genetic divergence is sometimes challenging \cite{bertolazzi2009learning}. In addition to being able to report likely required specimen sample sizes necessary to achieve saturation in species haplotype curves, it would be ideal if DNA barcoding studies could also provide a global measure of geographic dispersion to reliably test for cases of isolation-by-distance within species. Unfortunately, no such measure yet exists in this regard, making these kinds of analyses problematic. While model estimates may not be practical, having such a framework at hand that easily allows for the calculation of lower bounds for sample size offers researchers a glimpse into the most appropriate taxon sample sizes to target, and potentially where those taxa should be sampled. More crucially, the present simulation proposed herein can be employed to best determine the proper allocation of sampling effort, time and resources \cite{hortal2005ed}. Such work finds application in studies of metabarcoding \cite{wares2015can} as well as more broadly to global climate change \cite{pfenninger2012methodological}.



The development of a computational simulation of haplotype accumulation curves, a tool that can greatly aid biodiversity scientists in targeting species that will benefit from increased sampling effort, can be employed to build and grow BOLD with statistically defensible taxon records, which ultimately will allow more reliable specimen identification. This work is crucial because many taxon records currently in BOLD are known from only single specimens. Further, such a simulation algorithm could aid in species discovery through providing more reliable estimates of intraspecific sample sizes used in the \\ calculation of the barcode gap. Through developing statistically-relevant sample size \\ estimation tools that capture geographic and genetic variation within and between species, researchers will be able to improve sampling design strategies, which will lead to a better understanding (and improved database) of intra and interspecies genetic variation. As such, new methodologies will fill this void and contribute to the growing literature on sample size estimation for DNA barcoding.

\newpage

\section*{Acknowledgments}

We wish to greatly acknowledge the efforts of Rodger Gwiazdowski in providing \\ valuable edits to this manuscript. In addition, comments by Sarah (Sally) Adamowicz improved overall readability and flow of the manuscript considerably. Finally, two \\ anonymous revievers lent constructive feedback on this work, for which we are greatly appreciative.



This work was supported by a 2016/17 University of Guelph College of Physical and Engineering Science (CPES) Graduate Excellence Entrance Scholarship awarded to JDP.



The Dish With One Spoon Covenant speaks to our collective responsibility to steward and sustain the land and environment in which we live and work, so that all peoples, present and future, may benefit from the sustenance it provides. As we continue to strive to strengthen our relationships with and continue to learn from our Indigenous neighbours, we recognize the partnerships and knowledge that have guided the research conducted in our labs. We acknowledge that the University of Guelph resides in the ancestral and treaty lands of several Indigenous peoples, including the Attawandaron people and the Mississaugas of the Credit, and we recognize and honour our Anishinaabe, Haudenosaunee, and M{\'e}tis neighbours. We acknowledge that the work presented here has occurred on their traditional lands so that we might work to build lasting partnerships that respect, honour, and value the culture, traditions, and wisdom of those who have lived here since time immemorial.



\section*{Author Contributions}

JDP conducted the literature review and wrote the manuscript. DJG acted as an advisor in statistics. RHH acted as an advisor in DNA barcoding. All authors contributed to the revision of this manuscript and approved the final version. 



\section*{Conflict of Interest}

None declared.
